<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Jennifer Blanc" />


<title>Comparisons</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">infer_mutational_bias</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jgblanc/infer_mutational_bias">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Comparisons</h1>
<h4 class="author">Jennifer Blanc</h4>
<h4 class="date">2/10/2020</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-02-11
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>infer_mutational_bias/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.4.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20191108code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20191108)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20191108code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20191108)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomjgblancinfermutationalbiastreef28be1131a253c938a890792e6d90d49c77e3bb4targetblankf28be11a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/jgblanc/infer_mutational_bias/tree/f28be1131a253c938a890792e6d90d49c77e3bb4" target="_blank">f28be11</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomjgblancinfermutationalbiastreef28be1131a253c938a890792e6d90d49c77e3bb4targetblankf28be11a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    .snakemake/conda-archive/
    Ignored:    .snakemake/conda/
    Ignored:    .snakemake/locks/
    Ignored:    .snakemake/shadow/
    Ignored:    .snakemake/singularity/
    Ignored:    analysis/.DS_Store
    Ignored:    code/.DS_Store
    Ignored:    code/parse_gwas_atlas/.DS_Store
    Ignored:    data/.ipynb_checkpoints/
    Ignored:    data/47UKBB/
    Ignored:    data/BBJ/
    Ignored:    data/GWAS_ATLAS/
    Ignored:    data/UKBB/
    Ignored:    docs/.DS_Store
    Ignored:    docs/assets/.DS_Store
    Ignored:    output/.DS_Store
    Ignored:    output/47UKBB/
    Ignored:    output/GWAS_ATLAS/

Untracked files:
    Untracked:  .snakemake/log/2020-01-30T145351.841085.snakemake.log
    Untracked:  .snakemake/log/2020-01-30T152809.242567.snakemake.log
    Untracked:  .snakemake/log/2020-01-30T152814.933120.snakemake.log
    Untracked:  .snakemake/log/2020-01-30T161043.605380.snakemake.log
    Untracked:  .snakemake/log/2020-01-30T163643.702329.snakemake.log
    Untracked:  .snakemake/log/2020-01-30T163914.317308.snakemake.log
    Untracked:  .snakemake/log/2020-01-30T165942.174064.snakemake.log
    Untracked:  .snakemake/log/2020-01-31T132509.011706.snakemake.log
    Untracked:  .snakemake/log/2020-01-31T132531.112131.snakemake.log
    Untracked:  .snakemake/log/2020-01-31T135909.720564.snakemake.log
    Untracked:  .snakemake/log/2020-01-31T135932.405589.snakemake.log
    Untracked:  .snakemake/log/2020-01-31T140258.528215.snakemake.log
    Untracked:  .snakemake/log/2020-01-31T154259.739904.snakemake.log
    Untracked:  .snakemake/log/2020-01-31T161214.457782.snakemake.log
    Untracked:  .snakemake/log/2020-02-04T122755.245485.snakemake.log
    Untracked:  .snakemake/log/2020-02-04T125925.054486.snakemake.log
    Untracked:  .snakemake/log/2020-02-04T131309.070607.snakemake.log
    Untracked:  .snakemake/log/2020-02-04T165741.607714.snakemake.log
    Untracked:  .snakemake/log/2020-02-04T165832.953657.snakemake.log
    Untracked:  .snakemake/log/2020-02-04T165845.720203.snakemake.log
    Untracked:  .snakemake/log/2020-02-04T170106.750876.snakemake.log
    Untracked:  .snakemake/log/2020-02-04T173554.946756.snakemake.log
    Untracked:  .snakemake/log/2020-02-04T203921.856975.snakemake.log
    Untracked:  .snakemake/log/2020-02-04T203936.733129.snakemake.log
    Untracked:  .snakemake/log/2020-02-05T112056.885301.snakemake.log
    Untracked:  .snakemake/log/2020-02-05T112242.653088.snakemake.log
    Untracked:  .snakemake/log/2020-02-05T112254.668205.snakemake.log
    Untracked:  .snakemake/log/2020-02-05T112317.962078.snakemake.log
    Untracked:  .snakemake/log/2020-02-05T114409.152072.snakemake.log
    Untracked:  .snakemake/log/2020-02-05T115017.456859.snakemake.log
    Untracked:  .snakemake/log/2020-02-05T115225.611708.snakemake.log
    Untracked:  .snakemake/log/2020-02-05T131627.566541.snakemake.log
    Untracked:  .snakemake/log/2020-02-05T135142.429095.snakemake.log
    Untracked:  .snakemake/log/2020-02-05T135147.725230.snakemake.log
    Untracked:  .snakemake/log/2020-02-06T170947.682904.snakemake.log
    Untracked:  .snakemake/log/2020-02-06T170955.880903.snakemake.log
    Untracked:  .snakemake/log/2020-02-06T171020.234309.snakemake.log
    Untracked:  .snakemake/log/2020-02-06T171059.838074.snakemake.log
    Untracked:  .snakemake/log/2020-02-06T171138.722326.snakemake.log
    Untracked:  .snakemake/log/2020-02-06T172326.673867.snakemake.log
    Untracked:  .snakemake/log/2020-02-09T161433.704127.snakemake.log
    Untracked:  .snakemake/log/2020-02-09T161518.690570.snakemake.log
    Untracked:  .snakemake/log/2020-02-09T161701.456606.snakemake.log
    Untracked:  .snakemake/log/2020-02-09T161908.570560.snakemake.log
    Untracked:  .snakemake/log/2020-02-09T163531.063766.snakemake.log
    Untracked:  .snakemake/log/2020-02-09T164707.068682.snakemake.log
    Untracked:  .snakemake/log/2020-02-09T164713.498823.snakemake.log
    Untracked:  .snakemake/log/2020-02-09T164933.856759.snakemake.log
    Untracked:  .snakemake/log/2020-02-09T165246.623702.snakemake.log
    Untracked:  .snakemake/log/2020-02-09T165729.736217.snakemake.log
    Untracked:  .snakemake/log/2020-02-09T165753.884876.snakemake.log
    Untracked:  .snakemake/log/2020-02-09T165811.808410.snakemake.log
    Untracked:  .snakemake/log/2020-02-09T170118.509824.snakemake.log
    Untracked:  .snakemake/log/2020-02-09T185943.187754.snakemake.log
    Untracked:  .snakemake/log/2020-02-09T201412.006222.snakemake.log
    Untracked:  .snakemake/log/2020-02-09T201442.388739.snakemake.log
    Untracked:  .snakemake/log/2020-02-09T203655.598430.snakemake.log
    Untracked:  .snakemake/log/2020-02-10T123520.594963.snakemake.log
    Untracked:  .snakemake/log/2020-02-10T140913.222118.snakemake.log
    Untracked:  .snakemake/log/2020-02-10T141224.194342.snakemake.log
    Untracked:  .snakemake/log/2020-02-10T142516.378779.snakemake.log
    Untracked:  .snakemake/log/2020-02-10T150055.924168.snakemake.log
    Untracked:  .snakemake/log/2020-02-10T150106.868447.snakemake.log
    Untracked:  .snakemake/metadata/b3V0cHV0L0JCSi9wYXJzZWRfZ3dhcy9CTUlfMV9wYXJzZWQudHh0
    Untracked:  .snakemake/metadata/b3V0cHV0L0JCSi9wYXJzZWRfZ3dhcy9IRExfNDdfcGFyc2VkLnR4dA==
    Untracked:  .snakemake/metadata/b3V0cHV0L0JCSi9wYXJzZWRfZ3dhcy9IYl80OV9wYXJzZWQudHh0
    Untracked:  .snakemake/metadata/b3V0cHV0L0JCSi9wYXJzZWRfZ3dhcy9IZWlnaHRfMTM1X3BhcnNlZC50eHQ=
    Untracked:  .snakemake/metadata/b3V0cHV0L0JCSi9wYXJzZWRfZ3dhcy9MRExfNjFfcGFyc2VkLnR4dA==
    Untracked:  .snakemake/metadata/b3V0cHV0L0JCSi9wYXJzZWRfZ3dhcy9QbHRfOTdfcGFyc2VkLnR4dA==
    Untracked:  .snakemake/metadata/b3V0cHV0L0JCSi9wYXJzZWRfZ3dhcy9UR18xMDlfcGFyc2VkLnR4dA==
    Untracked:  .snakemake/metadata/b3V0cHV0L0dXQVNfQVRMQVMvZXZvX2FkZGVkL01TXzI0MDc2NjAyX2V2by50eHQ=
    Untracked:  .snakemake/metadata/b3V0cHV0L0dXQVNfQVRMQVMvZXZvX2FkZGVkL0FEXzQwOTRfZXZvLnR4dA==
    Untracked:  .snakemake/metadata/b3V0cHV0L0dXQVNfQVRMQVMvZXZvX2FkZGVkL0FPYXN0aG1hXzQyNjdfZXZvLnR4dA==
    Untracked:  .snakemake/metadata/b3V0cHV0L0dXQVNfQVRMQVMvZXZvX2FkZGVkL0JNSV80MDc0X2V2by50eHQ=
    Untracked:  .snakemake/metadata/b3V0cHV0L0dXQVNfQVRMQVMvZXZvX2FkZGVkL0NPYXN0aG1hXzQyNjZfZXZvLnR4dA==
    Untracked:  .snakemake/metadata/b3V0cHV0L0dXQVNfQVRMQVMvZXZvX2FkZGVkL0VBbWV0YV80MDY2X2V2by50eHQ=
    Untracked:  .snakemake/metadata/b3V0cHV0L0dXQVNfQVRMQVMvZXZvX2FkZGVkL0VBdWtiYl8zNDA5X2V2by50eHQ=
    Untracked:  .snakemake/metadata/b3V0cHV0L0dXQVNfQVRMQVMvZXZvX2FkZGVkL0lCRF8yMDMxX2V2by50eHQ=
    Untracked:  .snakemake/metadata/b3V0cHV0L0dXQVNfQVRMQVMvZXZvX2FkZGVkL1VDXzIwMzBfZXZvLnR4dA==
    Untracked:  .snakemake/metadata/b3V0cHV0L0dXQVNfQVRMQVMvZXZvX2FkZGVkL215b3BpYV8zNTM5X2V2by50eHQ=
    Untracked:  .snakemake/metadata/b3V0cHV0L0dXQVNfQVRMQVMvZXZvX2FkZGVkL2JyZWFzdGNhbmNlcl8zNjU5X2V2by50eHQ=
    Untracked:  .snakemake/metadata/b3V0cHV0L0dXQVNfQVRMQVMvZXZvX2FkZGVkL3BhcmtpbnNvbl80MTY3X2V2by50eHQ=
    Untracked:  .snakemake/metadata/b3V0cHV0L0dXQVNfQVRMQVMvZXZvX2FkZGVkL3ZpdGlsaWdvXzM4MzJfZXZvLnR4dA==
    Untracked:  .snakemake/metadata/b3V0cHV0L0dXQVNfQVRMQVMvcGFyc2VkX2d3YXMvQ09hc3RobWFfNDI2Nl9wYXJzZWQudHh0
    Untracked:  .snakemake/metadata/b3V0cHV0L0dXQVNfQVRMQVMvcGFyc2VkX2d3YXMvQU9hc3RobWFfNDI2N19wYXJzZWQudHh0
    Untracked:  .snakemake/metadata/b3V0cHV0L0dXQVNfQVRMQVMvcGFyc2VkX2d3YXMvQURfNDA5NF9wYXJzZWQudHh0
    Untracked:  .snakemake/metadata/b3V0cHV0L0dXQVNfQVRMQVMvcGFyc2VkX2d3YXMvQk1JXzQwNzRfcGFyc2VkLnR4dA==
    Untracked:  .snakemake/metadata/b3V0cHV0L0dXQVNfQVRMQVMvcGFyc2VkX2d3YXMvRUF1a2JiXzM0MDlfcGFyc2VkLnR4dA==
    Untracked:  .snakemake/metadata/b3V0cHV0L0dXQVNfQVRMQVMvcGFyc2VkX2d3YXMvRUFtZXRhXzQwNjZfcGFyc2VkLnR4dA==
    Untracked:  .snakemake/metadata/b3V0cHV0L0dXQVNfQVRMQVMvcGFyc2VkX2d3YXMvSUJEXzIwMzFfcGFyc2VkLnR4dA==
    Untracked:  .snakemake/metadata/b3V0cHV0L0dXQVNfQVRMQVMvcGFyc2VkX2d3YXMvTVNfMjQwNzY2MDJfcGFyc2VkLnR4dA==
    Untracked:  .snakemake/metadata/b3V0cHV0L0dXQVNfQVRMQVMvcGFyc2VkX2d3YXMvVUNfMjAzMF9wYXJzZWQudHh0
    Untracked:  .snakemake/metadata/b3V0cHV0L0dXQVNfQVRMQVMvcGFyc2VkX2d3YXMvbXlvcGlhXzM1MzlfcGFyc2VkLnR4dA==
    Untracked:  .snakemake/metadata/b3V0cHV0L0dXQVNfQVRMQVMvcGFyc2VkX2d3YXMvcGFya2luc29uXzQxNjdfcGFyc2VkLnR4dA==
    Untracked:  .snakemake/metadata/b3V0cHV0L1VLQkIvcGFyc2VkX2d3YXMvQk1JXzIzMTA0X3BhcnNlZC50eHQ=
    Untracked:  .snakemake/metadata/b3V0cHV0L1VLQkIvcGFyc2VkX2d3YXMvSERMXzMwNzYwX3BhcnNlZC50eHQ=
    Untracked:  .snakemake/metadata/b3V0cHV0L1VLQkIvcGFyc2VkX2d3YXMvSGJfMzAwNTBfcGFyc2VkLnR4dA==
    Untracked:  .snakemake/metadata/b3V0cHV0L1VLQkIvcGFyc2VkX2d3YXMvSGVpZ2h0XzUwX3BhcnNlZC50eHQ=
    Untracked:  .snakemake/metadata/b3V0cHV0L1VLQkIvcGFyc2VkX2d3YXMvTERMXzMwNzgwX3BhcnNlZC50eHQ=
    Untracked:  .snakemake/metadata/b3V0cHV0L1VLQkIvcGFyc2VkX2d3YXMvUGx0XzMwMDgwX3BhcnNlZC50eHQ=
    Untracked:  .snakemake/metadata/b3V0cHV0L1VLQkIvcGFyc2VkX2d3YXMvVEdfMzA4NzBfcGFyc2VkLnR4dA==
    Untracked:  code/Neal_UKBB.R
    Untracked:  code/parse_BBJ/
    Untracked:  code/parse_gwas_atlas/AD_4094.R
    Untracked:  code/parse_gwas_atlas/AOasthma_4267.R
    Untracked:  code/parse_gwas_atlas/BMI_4074.R
    Untracked:  code/parse_gwas_atlas/COasthma_4266.R
    Untracked:  code/parse_gwas_atlas/EAmeta_4066.R
    Untracked:  code/parse_gwas_atlas/EAukbb_3409.R
    Untracked:  code/parse_gwas_atlas/EAukbb_3509.R
    Untracked:  code/parse_gwas_atlas/IBD_2031.R
    Untracked:  code/parse_gwas_atlas/MS_24076602.R
    Untracked:  code/parse_gwas_atlas/UC_2030.R
    Untracked:  code/parse_gwas_atlas/myopia_3539.R
    Untracked:  code/parse_gwas_atlas/myopic_3539.R
    Untracked:  code/parse_gwas_atlas/parkinson_4167.R
    Untracked:  output/BBJ/
    Untracked:  output/UKBB/

Unstaged changes:
    Modified:   .snakemake/metadata/b3V0cHV0L0dXQVNfQVRMQVMvcGFyc2VkX2d3YXMvYnJlYXN0Y2FuY2VyXzM2NTlfcGFyc2VkLnR4dA==
    Modified:   .snakemake/metadata/b3V0cHV0L0dXQVNfQVRMQVMvcGFyc2VkX2d3YXMvdml0aWxpZ29fMzgzMl9wYXJzZWQudHh0
    Deleted:    code/#get_evolutionary_information_from_1kg_GWAS_ATLAS.py#
    Modified:   code/.ipynb_checkpoints/Untitled-checkpoint.ipynb
    Modified:   code/Untitled.ipynb
    Modified:   code/get_evolutionary_information_from_1kg_GWAS_ATLAS.py
    Modified:   code/get_evolutionary_information_from_1kg_GWAS_ATLAS.py~
    Modified:   code/parse_gwas_atlas/vitiligo_3832.R
    Modified:   snakefile
    Modified:   snakefile~

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jgblanc/infer_mutational_bias/blob/f28be1131a253c938a890792e6d90d49c77e3bb4/analysis/BBJvUKKBB.Rmd" target="_blank">f28be11</a>
</td>
<td>
jgblanc
</td>
<td>
2020-02-11
</td>
<td>
Did comparisons
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="intro" class="section level2">
<h2>Intro</h2>
<p>I downloaded all the BBJ summary statistics from (<a href="http://jenger.riken.jp/en/result" class="uri">http://jenger.riken.jp/en/result</a>). The ID number of the GWAS comes after the trait name in the file name. The UKBB summary statistics cam from the Neale lab (<a href="https://docs.google.com/spreadsheets/d/1kvPoupSzsSFBNSztMzl04xMoSC3Kcx3CrjVf4yBmESU/edit?ts=5b5f17db#gid=227859291" class="uri">https://docs.google.com/spreadsheets/d/1kvPoupSzsSFBNSztMzl04xMoSC3Kcx3CrjVf4yBmESU/edit?ts=5b5f17db#gid=227859291</a>). For height and BMI I used the “raw” files that hadn’t been transformed, for all the other traits I used the “irnt” files that used a inverse rank normal transformation. I used my own scripts to standardize the files and calculate the the effect allele frequecy. Here I loaded both the files for both biobanks and polorize effect size to be the BETA for the minor allele. I then binned the SNPs based on MAF to 100 bins with approximately equal numbers of SNPs. I then calculated the the average effect size per bin. Finally I plot the frequency (the max frequency of the bin) vs the mean effect size per bin. The main thing to note is that I haven’t LD clumped any of these summary statistics yet…I’m working on setting up an LD clumping pipeline using PLINK but haven’t finished yet.</p>
</div>
<div id="tratis" class="section level2">
<h2>Tratis</h2>
<pre class="r"><code>get_ES &lt;- function(df) {
  df$ES &lt;- NA
  minor_ef &lt;- subset(df,df$EAF == df$MAF)
  minor_ef$ES &lt;- minor_ef$BETA
  major_ef &lt;- subset(df, df$EAF != df$MAF)
  major_ef$ES &lt;- -1 * major_ef$BETA
  return(rbind(minor_ef,major_ef))
} 

bin_data &lt;- function(df) {
  df &lt;- df %&gt;% select(ES,MAF)
  df &lt;- df[order(MAF), ]
  df$BIN &lt;- as.numeric(cut2(df$MAF, g=100))
  return(df)
}

group_dat &lt;- function(df) {
  dat &lt;- df %&gt;% group_by(BIN) %&gt;% summarise(AVG = mean(ES), MAX = max(MAF))
}</code></pre>
<p><strong>Height</strong></p>
<pre class="r"><code>uk_height &lt;- group_dat(bin_data(get_ES(fread(&quot;../output/UKBB/parsed_gwas/Height_50_parsed.txt&quot;))))
jb_height &lt;- group_dat(bin_data(get_ES(fread(&quot;../output/BBJ/parsed_gwas/Height_135_parsed.txt&quot;))))

pl1 &lt;- ggplot(uk_height, aes(x=MAX, y=AVG)) + geom_point(color = &quot;darkviolet&quot;) + theme_bw() + xlab(&quot;Frequency&quot;) + ylab(&quot;Mean Effect Size&quot;) + ggtitle(&quot;UKBB&quot;) + theme(plot.title = element_text(hjust = 0.5)) 
pl2 &lt;- ggplot(jb_height, aes(x=MAX, y=AVG)) + geom_point(color= &quot;darkgreen&quot;) + theme_bw()+ xlab(&quot;Frequency&quot;) + ylab(&quot;Mean Effect Size&quot;) + ggtitle(&quot;BBJ&quot;) + theme(plot.title = element_text(hjust = 0.5)) 

ggarrange(pl1, pl2)</code></pre>
<p><img src="figure/BBJvUKKBB.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Remove the lowest frequnecy bin</p>
<pre class="r"><code>pl1 &lt;- ggplot(uk_height[-1,], aes(x=MAX, y=AVG)) + geom_point(color = &quot;darkviolet&quot;) + theme_bw() + xlab(&quot;Frequency&quot;) + ylab(&quot;Mean Effect Size&quot;) + ggtitle(&quot;UKBB&quot;) + theme(plot.title = element_text(hjust = 0.5)) 
pl2 &lt;- ggplot(jb_height[-1,], aes(x=MAX, y=AVG)) + geom_point(color= &quot;darkgreen&quot;) + theme_bw()+ xlab(&quot;Frequency&quot;) + ylab(&quot;Mean Effect Size&quot;) + ggtitle(&quot;BBJ&quot;) + theme(plot.title = element_text(hjust = 0.5)) 

ggarrange(pl1, pl2)</code></pre>
<p><img src="figure/BBJvUKKBB.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><strong>BMI</strong></p>
<pre class="r"><code>uk_BMI &lt;- group_dat(bin_data(get_ES(fread(&quot;../output/UKBB/parsed_gwas/BMI_23104_parsed.txt&quot;))))
jb_BMI &lt;- group_dat(bin_data(get_ES(fread(&quot;../output/BBJ/parsed_gwas/BMI_1_parsed.txt&quot;))))

pl1 &lt;- ggplot(uk_BMI, aes(x=MAX, y=AVG)) + geom_point(color = &quot;darkviolet&quot;) + theme_bw() + xlab(&quot;Frequency&quot;) + ylab(&quot;Mean Effect Size&quot;) + ggtitle(&quot;UKBB&quot;) + theme(plot.title = element_text(hjust = 0.5)) 
pl2 &lt;- ggplot(jb_BMI, aes(x=MAX, y=AVG)) + geom_point(color= &quot;darkgreen&quot;) + theme_bw()+ xlab(&quot;Frequency&quot;) + ylab(&quot;Mean Effect Size&quot;) + ggtitle(&quot;BBJ&quot;) + theme(plot.title = element_text(hjust = 0.5)) 

ggarrange(pl1, pl2)</code></pre>
<p><img src="figure/BBJvUKKBB.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Remove lowest frequency bin</p>
<pre class="r"><code>pl1 &lt;- ggplot(uk_BMI[-1,], aes(x=MAX, y=AVG)) + geom_point(color = &quot;darkviolet&quot;) + theme_bw() + xlab(&quot;Frequency&quot;) + ylab(&quot;Mean Effect Size&quot;) + ggtitle(&quot;UKBB&quot;) + theme(plot.title = element_text(hjust = 0.5)) 
pl2 &lt;- ggplot(jb_BMI[-1,], aes(x=MAX, y=AVG)) + geom_point(color= &quot;darkgreen&quot;) + theme_bw()+ xlab(&quot;Frequency&quot;) + ylab(&quot;Mean Effect Size&quot;) + ggtitle(&quot;BBJ&quot;) + theme(plot.title = element_text(hjust = 0.5)) 

ggarrange(pl1, pl2)</code></pre>
<p><img src="figure/BBJvUKKBB.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><strong>HDL</strong></p>
<pre class="r"><code>uk_HDL &lt;- group_dat(bin_data(get_ES(fread(&quot;../output/UKBB/parsed_gwas/HDL_30760_parsed.txt&quot;))))
jb_HDL &lt;- group_dat(bin_data(get_ES(fread(&quot;../output/BBJ/parsed_gwas/HDL_47_parsed.txt&quot;))))

pl1 &lt;- ggplot(uk_HDL, aes(x=MAX, y=AVG)) + geom_point(color = &quot;darkviolet&quot;) + theme_bw() + xlab(&quot;Frequency&quot;) + ylab(&quot;Mean Effect Size&quot;) + ggtitle(&quot;UKBB&quot;) + theme(plot.title = element_text(hjust = 0.5)) 
pl2 &lt;- ggplot(jb_HDL, aes(x=MAX, y=AVG)) + geom_point(color= &quot;darkgreen&quot;) + theme_bw()+ xlab(&quot;Frequency&quot;) + ylab(&quot;Mean Effect Size&quot;) + ggtitle(&quot;BBJ&quot;) + theme(plot.title = element_text(hjust = 0.5)) 

ggarrange(pl1, pl2)</code></pre>
<p><img src="figure/BBJvUKKBB.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Remove lowest frequency bin</p>
<pre class="r"><code>pl1 &lt;- ggplot(uk_HDL[-1,], aes(x=MAX, y=AVG)) + geom_point(color = &quot;darkviolet&quot;) + theme_bw() + xlab(&quot;Frequency&quot;) + ylab(&quot;Mean Effect Size&quot;) + ggtitle(&quot;UKBB&quot;) + theme(plot.title = element_text(hjust = 0.5)) 
pl2 &lt;- ggplot(jb_HDL[-1,], aes(x=MAX, y=AVG)) + geom_point(color= &quot;darkgreen&quot;) + theme_bw()+ xlab(&quot;Frequency&quot;) + ylab(&quot;Mean Effect Size&quot;) + ggtitle(&quot;BBJ&quot;) + theme(plot.title = element_text(hjust = 0.5)) 

ggarrange(pl1, pl2)</code></pre>
<p><img src="figure/BBJvUKKBB.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><strong>LDL</strong></p>
<pre class="r"><code>uk_LDL &lt;- group_dat(bin_data(get_ES(fread(&quot;../output/UKBB/parsed_gwas/LDL_30780_parsed.txt&quot;))))
jb_LDL &lt;- group_dat(bin_data(get_ES(fread(&quot;../output/BBJ/parsed_gwas/LDL_61_parsed.txt&quot;))))

pl1 &lt;- ggplot(uk_LDL, aes(x=MAX, y=AVG)) + geom_point(color = &quot;darkviolet&quot;) + theme_bw() + xlab(&quot;Frequency&quot;) + ylab(&quot;Mean Effect Size&quot;) + ggtitle(&quot;UKBB&quot;) + theme(plot.title = element_text(hjust = 0.5)) 
pl2 &lt;- ggplot(jb_LDL, aes(x=MAX, y=AVG)) + geom_point(color= &quot;darkgreen&quot;) + theme_bw()+ xlab(&quot;Frequency&quot;) + ylab(&quot;Mean Effect Size&quot;) + ggtitle(&quot;BBJ&quot;) + theme(plot.title = element_text(hjust = 0.5)) 

ggarrange(pl1, pl2)</code></pre>
<p><img src="figure/BBJvUKKBB.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Remove lowest frequency bin</p>
<pre class="r"><code>pl1 &lt;- ggplot(uk_LDL[-1,], aes(x=MAX, y=AVG)) + geom_point(color = &quot;darkviolet&quot;) + theme_bw() + xlab(&quot;Frequency&quot;) + ylab(&quot;Mean Effect Size&quot;) + ggtitle(&quot;UKBB&quot;) + theme(plot.title = element_text(hjust = 0.5)) 
pl2 &lt;- ggplot(jb_LDL[-1,], aes(x=MAX, y=AVG)) + geom_point(color= &quot;darkgreen&quot;) + theme_bw()+ xlab(&quot;Frequency&quot;) + ylab(&quot;Mean Effect Size&quot;) + ggtitle(&quot;BBJ&quot;) + theme(plot.title = element_text(hjust = 0.5)) 

ggarrange(pl1, pl2)</code></pre>
<p><img src="figure/BBJvUKKBB.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><strong>Triglycerides</strong></p>
<pre class="r"><code>uk_TG &lt;- group_dat(bin_data(get_ES(fread(&quot;../output/UKBB/parsed_gwas/TG_30870_parsed.txt&quot;))))
jb_TG &lt;- group_dat(bin_data(get_ES(fread(&quot;../output/BBJ/parsed_gwas/TG_109_parsed.txt&quot;))))

pl1 &lt;- ggplot(uk_TG, aes(x=MAX, y=AVG)) + geom_point(color = &quot;darkviolet&quot;) + theme_bw() + xlab(&quot;Frequency&quot;) + ylab(&quot;Mean Effect Size&quot;) + ggtitle(&quot;UKBB&quot;) + theme(plot.title = element_text(hjust = 0.5)) 
pl2 &lt;- ggplot(jb_TG, aes(x=MAX, y=AVG)) + geom_point(color= &quot;darkgreen&quot;) + theme_bw()+ xlab(&quot;Frequency&quot;) + ylab(&quot;Mean Effect Size&quot;) + ggtitle(&quot;BBJ&quot;) + theme(plot.title = element_text(hjust = 0.5)) 

ggarrange(pl1, pl2)</code></pre>
<p><img src="figure/BBJvUKKBB.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Remove lowest frequency bin</p>
<pre class="r"><code>pl1 &lt;- ggplot(uk_TG[-1,], aes(x=MAX, y=AVG)) + geom_point(color = &quot;darkviolet&quot;) + theme_bw() + xlab(&quot;Frequency&quot;) + ylab(&quot;Mean Effect Size&quot;) + ggtitle(&quot;UKBB&quot;) + theme(plot.title = element_text(hjust = 0.5)) 
pl2 &lt;- ggplot(jb_TG[-1,], aes(x=MAX, y=AVG)) + geom_point(color= &quot;darkgreen&quot;) + theme_bw()+ xlab(&quot;Frequency&quot;) + ylab(&quot;Mean Effect Size&quot;) + ggtitle(&quot;BBJ&quot;) + theme(plot.title = element_text(hjust = 0.5)) 

ggarrange(pl1, pl2)</code></pre>
<p><img src="figure/BBJvUKKBB.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><strong>Hemoglobin</strong></p>
<pre class="r"><code>uk_Hb &lt;- group_dat(bin_data(get_ES(fread(&quot;../output/UKBB/parsed_gwas/Hb_30050_parsed.txt&quot;))))
jb_Hb &lt;- group_dat(bin_data(get_ES(fread(&quot;../output/BBJ/parsed_gwas/Hb_49_parsed.txt&quot;))))

pl1 &lt;- ggplot(uk_Hb, aes(x=MAX, y=AVG)) + geom_point(color = &quot;darkviolet&quot;) + theme_bw() + xlab(&quot;Frequency&quot;) + ylab(&quot;Mean Effect Size&quot;) + ggtitle(&quot;UKBB&quot;) + theme(plot.title = element_text(hjust = 0.5)) 
pl2 &lt;- ggplot(jb_Hb, aes(x=MAX, y=AVG)) + geom_point(color= &quot;darkgreen&quot;) + theme_bw()+ xlab(&quot;Frequency&quot;) + ylab(&quot;Mean Effect Size&quot;) + ggtitle(&quot;BBJ&quot;) + theme(plot.title = element_text(hjust = 0.5)) 

ggarrange(pl1, pl2)</code></pre>
<p><img src="figure/BBJvUKKBB.Rmd/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Remove lowest frequency bin</p>
<pre class="r"><code>pl1 &lt;- ggplot(uk_Hb[-1,], aes(x=MAX, y=AVG)) + geom_point(color = &quot;darkviolet&quot;) + theme_bw() + xlab(&quot;Frequency&quot;) + ylab(&quot;Mean Effect Size&quot;) + ggtitle(&quot;UKBB&quot;) + theme(plot.title = element_text(hjust = 0.5)) 
pl2 &lt;- ggplot(jb_Hb[-1,], aes(x=MAX, y=AVG)) + geom_point(color= &quot;darkgreen&quot;) + theme_bw()+ xlab(&quot;Frequency&quot;) + ylab(&quot;Mean Effect Size&quot;) + ggtitle(&quot;BBJ&quot;) + theme(plot.title = element_text(hjust = 0.5)) 

ggarrange(pl1, pl2)</code></pre>
<p><img src="figure/BBJvUKKBB.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><strong>Platelet Count</strong></p>
<pre class="r"><code>uk_Plt &lt;- group_dat(bin_data(get_ES(fread(&quot;../output/UKBB/parsed_gwas/Plt_30080_parsed.txt&quot;))))
jb_Plt &lt;- group_dat(bin_data(get_ES(fread(&quot;../output/BBJ/parsed_gwas/Plt_97_parsed.txt&quot;))))

pl1 &lt;- ggplot(uk_Plt, aes(x=MAX, y=AVG)) + geom_point(color = &quot;darkviolet&quot;) + theme_bw() + xlab(&quot;Frequency&quot;) + ylab(&quot;Mean Effect Size&quot;) + ggtitle(&quot;UKBB&quot;) + theme(plot.title = element_text(hjust = 0.5)) 
pl2 &lt;- ggplot(jb_Plt, aes(x=MAX, y=AVG)) + geom_point(color= &quot;darkgreen&quot;) + theme_bw()+ xlab(&quot;Frequency&quot;) + ylab(&quot;Mean Effect Size&quot;) + ggtitle(&quot;BBJ&quot;) + theme(plot.title = element_text(hjust = 0.5)) 

ggarrange(pl1, pl2)</code></pre>
<p><img src="figure/BBJvUKKBB.Rmd/unnamed-chunk-14-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Remove lowest frequency bin</p>
<pre class="r"><code>pl1 &lt;- ggplot(uk_Plt[-1,], aes(x=MAX, y=AVG)) + geom_point(color = &quot;darkviolet&quot;) + theme_bw() + xlab(&quot;Frequency&quot;) + ylab(&quot;Mean Effect Size&quot;) + ggtitle(&quot;UKBB&quot;) + theme(plot.title = element_text(hjust = 0.5)) 
pl2 &lt;- ggplot(jb_Plt[-1,], aes(x=MAX, y=AVG)) + geom_point(color= &quot;darkgreen&quot;) + theme_bw()+ xlab(&quot;Frequency&quot;) + ylab(&quot;Mean Effect Size&quot;) + ggtitle(&quot;BBJ&quot;) + theme(plot.title = element_text(hjust = 0.5)) 

ggarrange(pl1, pl2)</code></pre>
<p><img src="figure/BBJvUKKBB.Rmd/unnamed-chunk-15-1.png" width="672" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.5.1 (2018-07-02)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS High Sierra 10.13.6

Matrix products: default
BLAS: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] Hmisc_4.2-0       Formula_1.2-3     survival_2.44-1.1
 [4] lattice_0.20-38   latex2exp_0.4.0   ggpubr_0.2       
 [7] magrittr_1.5      dplyr_0.8.1       ggplot2_3.2.1    
[10] data.table_1.12.2

loaded via a namespace (and not attached):
 [1] tidyselect_0.2.5    xfun_0.7            purrr_0.3.2        
 [4] splines_3.5.1       colorspace_1.4-1    htmltools_0.3.6    
 [7] yaml_2.2.0          base64enc_0.1-3     rlang_0.4.1        
[10] pillar_1.4.2        foreign_0.8-71      glue_1.3.1         
[13] withr_2.1.2         RColorBrewer_1.1-2  lifecycle_0.1.0    
[16] stringr_1.4.0       munsell_0.5.0       gtable_0.3.0       
[19] workflowr_1.4.0     htmlwidgets_1.3     evaluate_0.14      
[22] labeling_0.3        latticeExtra_0.6-28 knitr_1.23         
[25] htmlTable_1.13.1    Rcpp_1.0.3          acepack_1.4.1      
[28] scales_1.1.0        backports_1.1.5     checkmate_1.9.4    
[31] farver_2.0.1        fs_1.3.1            gridExtra_2.3      
[34] digest_0.6.22       stringi_1.4.3       grid_3.5.1         
[37] rprojroot_1.3-2     cowplot_0.9.4       tools_3.5.1        
[40] lazyeval_0.2.2      tibble_2.1.3        cluster_2.0.9      
[43] crayon_1.3.4        whisker_0.3-2       pkgconfig_2.0.3    
[46] Matrix_1.2-17       assertthat_0.2.1    rmarkdown_1.13     
[49] rstudioapi_0.10     R6_2.4.1            rpart_4.1-15       
[52] nnet_7.3-12         git2r_0.25.2        compiler_3.5.1     </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
