<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Jennifer Blanc" />


<title>misc_gwas</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">infer_mutational_bias</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jgblanc/infer_mutational_bias">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">misc_gwas</h1>
<h4 class="author">Jennifer Blanc</h4>
<h4 class="date">6/8/2020</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-06-29
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>infer_mutational_bias/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20191108code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20191108)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20191108code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20191108)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomjgblancinfermutationalbiastreeef2f9ef996162f407b507433df30b63d8e48416btargetblankef2f9efa"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/jgblanc/infer_mutational_bias/tree/ef2f9ef996162f407b507433df30b63d8e48416b" target="_blank">ef2f9ef</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomjgblancinfermutationalbiastreeef2f9ef996162f407b507433df30b63d8e48416btargetblankef2f9efa" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/jgblanc/infer_mutational_bias/tree/ef2f9ef996162f407b507433df30b63d8e48416b" target="_blank">ef2f9ef</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    .snakemake/conda-archive/
    Ignored:    .snakemake/conda/
    Ignored:    .snakemake/locks/
    Ignored:    .snakemake/shadow/
    Ignored:    .snakemake/singularity/
    Ignored:    analysis/.DS_Store
    Ignored:    code/.DS_Store
    Ignored:    code/parse_gwas_atlas/.DS_Store
    Ignored:    code/parse_gwas_atlas_nonsig/.DS_Store
    Ignored:    data/.ipynb_checkpoints/
    Ignored:    data/1000G_20101123_v3_GIANT_chr1_23_minimacnamesifnotRS_CEU_MAF0.01/
    Ignored:    data/ALL.chr1.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.EUR/
    Ignored:    data/GWAS_ATLAS/
    Ignored:    data/LD_Blocks/
    Ignored:    data/MISC_GWAS/
    Ignored:    data/STRAT/
    Ignored:    data/UKBB/
    Ignored:    output/.DS_Store
    Ignored:    output/47UKBB/
    Ignored:    output/BBJ/.DS_Store
    Ignored:    output/GWAS_ATLAS/
    Ignored:    output/STRAT/
    Ignored:    output/UKBB/.DS_Store

Untracked files:
    Untracked:  .snakemake/log/2020-03-23T164719.122778.snakemake.log
    Untracked:  .snakemake/log/2020-03-23T164841.172341.snakemake.log
    Untracked:  .snakemake/log/2020-03-23T165052.235224.snakemake.log
    Untracked:  .snakemake/log/2020-03-23T173201.671245.snakemake.log
    Untracked:  .snakemake/log/2020-03-23T190924.512591.snakemake.log
    Untracked:  .snakemake/log/2020-03-23T194341.864454.snakemake.log
    Untracked:  .snakemake/log/2020-03-24T091905.302205.snakemake.log
    Untracked:  .snakemake/log/2020-03-24T092201.882781.snakemake.log
    Untracked:  .snakemake/log/2020-03-24T092404.171989.snakemake.log
    Untracked:  .snakemake/log/2020-03-24T092428.802361.snakemake.log
    Untracked:  .snakemake/log/2020-03-24T093017.568005.snakemake.log
    Untracked:  .snakemake/log/2020-03-24T093241.050903.snakemake.log
    Untracked:  .snakemake/log/2020-03-24T093254.384314.snakemake.log
    Untracked:  .snakemake/log/2020-03-24T093258.136407.snakemake.log
    Untracked:  .snakemake/log/2020-03-24T093449.542770.snakemake.log
    Untracked:  .snakemake/log/2020-03-24T111104.620227.snakemake.log
    Untracked:  .snakemake/log/2020-03-24T111633.863041.snakemake.log
    Untracked:  .snakemake/log/2020-03-24T111701.823844.snakemake.log
    Untracked:  .snakemake/log/2020-03-24T152044.472618.snakemake.log
    Untracked:  .snakemake/log/2020-03-24T152202.883346.snakemake.log
    Untracked:  .snakemake/log/2020-03-24T152218.356089.snakemake.log
    Untracked:  .snakemake/log/2020-03-24T152224.553347.snakemake.log
    Untracked:  .snakemake/log/2020-03-24T152253.175681.snakemake.log
    Untracked:  .snakemake/log/2020-03-24T152612.976687.snakemake.log
    Untracked:  .snakemake/log/2020-03-24T152904.986735.snakemake.log
    Untracked:  .snakemake/log/2020-03-24T154155.249321.snakemake.log
    Untracked:  .snakemake/log/2020-03-24T154220.752872.snakemake.log
    Untracked:  .snakemake/log/2020-03-24T154225.402696.snakemake.log
    Untracked:  .snakemake/log/2020-03-24T154242.303180.snakemake.log
    Untracked:  .snakemake/log/2020-03-24T154540.706170.snakemake.log
    Untracked:  .snakemake/log/2020-03-24T154744.109794.snakemake.log
    Untracked:  .snakemake/log/2020-03-24T154756.043621.snakemake.log
    Untracked:  .snakemake/log/2020-03-24T155047.465119.snakemake.log
    Untracked:  .snakemake/log/2020-03-24T155100.393951.snakemake.log
    Untracked:  .snakemake/log/2020-03-24T155721.217180.snakemake.log
    Untracked:  .snakemake/log/2020-03-24T155802.251555.snakemake.log
    Untracked:  .snakemake/log/2020-03-24T155823.527017.snakemake.log
    Untracked:  .snakemake/log/2020-03-24T160158.325053.snakemake.log
    Untracked:  .snakemake/log/2020-03-24T160554.632453.snakemake.log
    Untracked:  .snakemake/log/2020-03-24T160813.358275.snakemake.log
    Untracked:  .snakemake/log/2020-03-24T161022.670871.snakemake.log
    Untracked:  .snakemake/log/2020-03-24T161257.193338.snakemake.log
    Untracked:  .snakemake/log/2020-03-30T171337.338623.snakemake.log
    Untracked:  .snakemake/metadata/b3V0cHV0L0dXQVNfQVRMQVMvZXZvX2FkZGVkL0NBRF8zOTI1LTAuMDAwMV9ldm8udHh0
    Untracked:  .snakemake/metadata/b3V0cHV0L0dXQVNfQVRMQVMvZXZvX2FkZGVkL1QyRF80MDg1LTAuMDAwMV9ldm8udHh0
    Untracked:  .snakemake/metadata/b3V0cHV0L0dXQVNfQVRMQVMvZXZvX2FkZGVkL1VDXzIwMzAtMC4wMDAxX2V2by50eHQ=
    Untracked:  .snakemake/metadata/b3V0cHV0L0dXQVNfQVRMQVMvZXZvX2FkZGVkL2hlaWdodF8xNDItMC4wMDAxX2V2by50eHQ=
    Untracked:  .snakemake/metadata/b3V0cHV0L0dXQVNfQVRMQVMvZXZvX2FkZGVkL3NjaGl6b3BocmVuaWFfMzk4Mi0wLjAwMDFfZXZvLnR4dA==
    Untracked:  .snakemake/metadata/b3V0cHV0L0dXQVNfQVRMQVMvcGFyc2VkX2d3YXMvQ0FEXzM5MjVfMC4wMDAxX3BhcnNlZC50eHQ=
    Untracked:  .snakemake/metadata/b3V0cHV0L0dXQVNfQVRMQVMvcGFyc2VkX2d3YXMvVDJEXzQwODVfMC4wMDAxX3BhcnNlZC50eHQ=
    Untracked:  .snakemake/metadata/b3V0cHV0L0dXQVNfQVRMQVMvcGFyc2VkX2d3YXMvVUNfMjAzMF8wLjAwMDFfcGFyc2VkLnR4dA==
    Untracked:  .snakemake/metadata/b3V0cHV0L0dXQVNfQVRMQVMvcGFyc2VkX2d3YXMvaGVpZ2h0XzE0Ml8wLjAwMDFfcGFyc2VkLnR4dA==
    Untracked:  .snakemake/metadata/b3V0cHV0L0dXQVNfQVRMQVMvcGFyc2VkX2d3YXMvc2NoaXpvcGhyZW5pYV8zOTgyXzAuMDAwMV9wYXJzZWQudHh0
    Untracked:  .snakemake/metadata/b3V0cHV0L0dXQVNfQVRMQVMvcHJ1bmVkL0NBRF8zOTI1XzAuMDAwMS5wcnVuZS5pbg==
    Untracked:  .snakemake/metadata/b3V0cHV0L0dXQVNfQVRMQVMvcHJ1bmVkL1QyRF80MDg1XzAuMDAwMS5wcnVuZS5pbg==
    Untracked:  .snakemake/metadata/b3V0cHV0L0dXQVNfQVRMQVMvcHJ1bmVkL1VDXzIwMzBfMC4wMDAxLnBydW5lLmlu
    Untracked:  .snakemake/metadata/b3V0cHV0L0dXQVNfQVRMQVMvcHJ1bmVkL2hlaWdodF8xNDJfMC4wMDAxLnBydW5lLmlu
    Untracked:  .snakemake/metadata/b3V0cHV0L0dXQVNfQVRMQVMvcHJ1bmVkL3NjaGl6b3BocmVuaWFfMzk4Ml8wLjAwMDEucHJ1bmUuaW4=
    Untracked:  .snakemake/scripts/tmp7997be50.get_evolutionary_information_from_1kg_GWAS_ATLAS.py
    Untracked:  Untitled.Rmd
    Untracked:  analysis/GWAS_Atlas_LD_Blocks.Rmd
    Untracked:  analysis/Instrumental_variable.Rmd
    Untracked:  analysis/LD_Block_Jacknife.Rmd
    Untracked:  analysis/pan_uk_results.Rmd
    Untracked:  analysis/test.Rmd
    Untracked:  code/LD_Block/
    Untracked:  code/parse_gwas_atlas_nonsig/height_142.R
    Untracked:  code/pc_regression.rds
    Untracked:  code/pc_regression.stan
    Untracked:  code/pc_regression_da.rds
    Untracked:  code/pc_regression_da.stan
    Untracked:  code/pc_regression_gwas.R
    Untracked:  code/pc_regression_parameters.R
    Untracked:  code/rsID_Pos_conversion.py
    Untracked:  code/rsID_Pos_conversion.py~
    Untracked:  output/GWAS_ATLAS_ANNOTATED/
    Untracked:  output/PC_REG/

Unstaged changes:
    Modified:   .snakemake/metadata/b3V0cHV0L1NUUkFUL3BydW5lZC9jaHIxX0VVUl8wLjA1LmVpZ2VudmVjLnZhci5EQS5wcnVuZS5pbg==
    Modified:   .snakemake/metadata/b3V0cHV0L1NUUkFUL3BydW5lZC9jaHIxX0VVUl8wLjAxLmVpZ2VudmVjLnZhci5EQS5wcnVuZS5pbg==
    Modified:   .snakemake/metadata/b3V0cHV0L1NUUkFUL3BydW5lZC9jaHIxX0VVUl8wLjEuZWlnZW52ZWMudmFyLkRBLnBydW5lLmlu
    Modified:   .snakemake/metadata/b3V0cHV0L1NUUkFUL3BydW5lZC9jaHIxX0VVUl8wLjIuZWlnZW52ZWMudmFyLkRBLnBydW5lLmlu
    Modified:   .snakemake/metadata/b3V0cHV0L1NUUkFUL3BydW5lZC9jaHIxX0VVUl8wLjMuZWlnZW52ZWMudmFyLkRBLnBydW5lLmlu
    Modified:   .snakemake/metadata/b3V0cHV0L1NUUkFUL3BydW5lZC9jaHIxX0VVUl8wLjQuZWlnZW52ZWMudmFyLkRBLnBydW5lLmlu
    Modified:   analysis/GWAS_ATLAS_results.Rmd
    Modified:   analysis/Stratification.Rmd
    Modified:   analysis/shift_GWAS_ATLAS.Rmd
    Modified:   analysis/strat_GWASs.Rmd
    Modified:   code/Neal_UKBB.R
    Deleted:    code/Untitled.ipynb
    Deleted:    code/make_DA_dict.py
    Deleted:    code/make_DA_dict.py~
    Modified:   code/parse_gwas_atlas/schizophrenia_3982.R
    Modified:   code/parse_gwas_atlas/schizophrenia_PGC.R
    Modified:   code/parse_gwas_atlas_nonsig/CAD_3925.R
    Modified:   code/parse_gwas_atlas_nonsig/T2D_4085.R
    Modified:   code/parse_gwas_atlas_nonsig/UC_2030.R
    Modified:   code/parse_gwas_atlas_nonsig/schizophrenia_3982.R
    Modified:   snakefile
    Modified:   snakefile~

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/misc_gwas.Rmd</code>) and HTML (<code>docs/misc_gwas.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jgblanc/infer_mutational_bias/blob/ef2f9ef996162f407b507433df30b63d8e48416b/analysis/misc_gwas.Rmd" target="_blank">ef2f9ef</a>
</td>
<td>
jgblanc
</td>
<td>
2020-06-29
</td>
<td>
added SCZ gwas
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jgblanc/infer_mutational_bias/845a11ed8974fff944ca18efb108928918ad9a44/docs/misc_gwas.html" target="_blank">845a11e</a>
</td>
<td>
jgblanc
</td>
<td>
2020-06-09
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jgblanc/infer_mutational_bias/blob/7e3297a0f6f8737751a6ddca54983a002c99ab77/analysis/misc_gwas.Rmd" target="_blank">7e3297a</a>
</td>
<td>
jgblanc
</td>
<td>
2020-06-09
</td>
<td>
start misc process
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="code" class="section level2">
<h2>Code</h2>
<p>Read in summary statistics</p>
<pre class="r"><code>read_ss &lt;- function(name) {
  df &lt;- fread(paste0(&quot;../data/MISC_GWAS/&quot;,name, &quot;.txt&quot;))
  return(df)
}</code></pre>
<p>Pick only GWAS Significant SNPs</p>
<pre class="r"><code>sig_snps &lt;- function(df, threshold) {
  out &lt;- df %&gt;% filter(P &lt; threshold)
  return(out)
}</code></pre>
<p>Pick lowest p-value SNP per block</p>
<pre class="r"><code>lowest_pval &lt;- function(df) {
  out &lt;- df %&gt;% group_by(LD_BLOCK) %&gt;% arrange(P) %&gt;% slice(1)
  return(out)
}</code></pre>
<p>Pick SNPs below a given threshold</p>
<pre class="r"><code>nonsig_snps &lt;- function(df, threshold) {
  out &lt;- df %&gt;% filter(abs(ES) &lt; threshold)
}</code></pre>
<p>Proportion of derived alleles that increase the trait</p>
<pre class="r"><code>increasing_derived_alleles &lt;- function(df) {
  return(sum(df$ES_DERIVED &gt; 0)/nrow(df))
}</code></pre>
<p>Proportion of derived alleles that increase the trait for multiple thresholds</p>
<pre class="r"><code>increasing_derived_alleles_thresolds &lt;- function(data) {
  df &lt;- matrix(NA, nrow = 5, ncol = 2)
  colnames(df) &lt;- c(&quot;Prop_Inc&quot;, &quot;Num_SNPs&quot;)
  threshold &lt;- c(1e-2, 1e-3, 1e-4, 1e-5, 1e-6)
  for (i in 1:5) {
    non_sig &lt;- nonsig_snps(data, threshold[i])
    df[i,1] &lt;- increasing_derived_alleles(non_sig)
    df[i,2] &lt;- nrow(non_sig)
    data &lt;- non_sig
  }
  return(df)
}</code></pre>
<p>Make ES vs IAF plot</p>
<pre class="r"><code>derive &lt;- c(&quot;springgreen4&quot;, &quot;purple1&quot;)
make_plot &lt;- function(df, name) {
  pl &lt;- ggplot(data=df, aes(x=IAF,y=abs(ES), color = INCREASING_DERIVED)) + geom_point(alpha=0.5, size=2) + theme_classic() + ggtitle(name) + labs(y = &quot;Effect Size&quot;, x = &quot;Trait Increaseing Allele Frequency&quot;,color = &quot;Risk Derived&quot;)+ ylim(0,max(abs(df$ES)+0.01)) +  xlim(0,1) + scale_color_manual(values = derive) + theme(plot.title = element_text(hjust = 0.5, size = 20), axis.text.x = element_text(size = 10), axis.text.y = element_text(size = 10), axis.title.x = element_text(size = 18), axis.title.y = element_text(size = 18), legend.title = element_text(size = 14), legend.text = element_text(size = 14), legend.position = &quot;bottom&quot;) 
  return(pl)
}</code></pre>
</div>
<div id="workflow-outline" class="section level2">
<h2>Workflow outline</h2>
<ol style="list-style-type: decimal">
<li>Download raw summary statistics<br />
</li>
<li>Parse all summary stats into same format: “SNP” (CHR:BP), “EA” (effect allele),“NEA”(non-effect allele), “CHR”, “BP”, “BETA” (Should be called OR-need to fix), “SE”, “P”, “EAF” (effect allele frequency) ,“MAF”, “IAF” (increasing allele frequency), “INCREASING_ALLELE” (BETA &gt; 0), “INCREASING” (Allele w/ BETA &gt; 0), “ES” (log(OR))</li>
<li>Add column that uses a table that has derived/ancestral assignments for 1KG SNPs to add the columns: “EA_DERIVED” (T/F effect allele derived), “DAF” (derived allele freq), “INCREASING_DERIVED” (T/F is the increasing allele derived), “ES_DERIVED” (Effect size of the derived allele). SNPs that don’t have derived information are dropped from the table.<br />
</li>
<li>Add column that assigns each SNP to one of 1703 LD Blocks defined by <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4731402/" class="uri">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4731402/</a></li>
</ol>
</div>
<div id="schizophrenia-gwas-results" class="section level2">
<h2>Schizophrenia GWAS Results</h2>
<p><strong>EUR Psychiatric Genetic Consortium</strong></p>
<p>The raw GWAS was downloaded from (<a href="https://www.med.unc.edu/pgc/download-results/scz/" class="uri">https://www.med.unc.edu/pgc/download-results/scz/</a>), EUR only descent. They are discussed in this paper: <a href="https://www.nature.com/articles/s41588-019-0512-x" class="uri">https://www.nature.com/articles/s41588-019-0512-x</a></p>
<p>First, we can plot the trait increasing allele frequency vs effect size for GWAS significant SNPs (P &lt; 5e-8), picking the lowest p-value per LD Block and coloring by derived status.</p>
<pre class="r"><code>data &lt;- fread(&quot;~/scratch/SCZ-EUR_PGC.txt&quot;)
sig &lt;- sig_snps(data, 5e-8)
ld_block &lt;- lowest_pval(sig)
make_plot(ld_block, &quot;SCZ-EUR_PGC&quot;)</code></pre>
<p><img src="figure/misc_gwas.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-1">
Past versions of unnamed-chunk-8-1.png
</button>
</p>
<div id="fig-unnamed-chunk-8-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/jgblanc/infer_mutational_bias/blob/845a11ed8974fff944ca18efb108928918ad9a44/docs/figure/misc_gwas.Rmd/unnamed-chunk-8-1.png" target="_blank">845a11e</a>
</td>
<td>
jgblanc
</td>
<td>
2020-06-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Now we can make the same plot but dropping all SNPs on chromosome 6.</p>
<pre class="r"><code>drop_six &lt;- ld_block %&gt;% filter(CHR != 6)
make_plot(drop_six, &quot;Exclude CHR 6&quot;)</code></pre>
<p><img src="figure/misc_gwas.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>We can also look at specific LD Blocks on chromosome 6</p>
<pre class="r"><code>p650 &lt;- data %&gt;% filter(CHR == 6 &amp; LD_BLOCK == 650) %&gt;% mutate(sig = P &lt; 5e-8) %&gt;%ggplot(aes(x = BP, y=ES_DERIVED, color = sig)) + geom_point()
p651 &lt;- data %&gt;% filter(CHR == 6 &amp; LD_BLOCK == 651) %&gt;% mutate(sig = P &lt; 5e-8) %&gt;%ggplot(aes(x = BP, y=ES_DERIVED, color = sig)) + geom_point()
p652 &lt;- data %&gt;% filter(CHR == 6 &amp; LD_BLOCK == 652) %&gt;% mutate(sig = P &lt; 5e-8) %&gt;%ggplot(aes(x = BP, y=ES_DERIVED, color = sig)) + geom_point()
p653 &lt;- data %&gt;% filter(CHR == 6 &amp; LD_BLOCK == 653) %&gt;% mutate(sig = P &lt; 5e-8) %&gt;%ggplot(aes(x = BP, y=ES_DERIVED, color = sig)) + geom_point()
ggarrange(p650,p651,p652,p653)</code></pre>
<p><img src="figure/misc_gwas.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-10-1">
Past versions of unnamed-chunk-10-1.png
</button>
</p>
<div id="fig-unnamed-chunk-10-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/jgblanc/infer_mutational_bias/blob/845a11ed8974fff944ca18efb108928918ad9a44/docs/figure/misc_gwas.Rmd/unnamed-chunk-10-1.png" target="_blank">845a11e</a>
</td>
<td>
jgblanc
</td>
<td>
2020-06-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#df &lt;- data %&gt;% mutate(daf_bin = cut(DAF,c(-Inf,0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9, Inf), labels = seq(0.1,1, 0.1))) %&gt;% mutate(es_bin = cut(ES_DERIVED,c(-Inf,-0.7, -0.6, -0.5, -0.4, -0.3, -0.2, -0.1, 0, 0.1,0.2,0.3,0.4,0.5,0.6,0.7, Inf), labels = c(-0.7, -0.6, -0.5, -0.4, -0.3, -0.2, -0.1, 0, 0.1,0.2,0.3,0.4,0.5,0.6,0.7,0.8)))
#df_bin &lt;- df %&gt;% group_by(daf_bin, es_bin) %&gt;% summarise(num_snps = n())
#ggplot(data = df_bin, aes(x=daf_bin, y=es_bin, fill = num_snps)) + geom_tile()</code></pre>
<p><strong>EAS Psychiatric Genetic Consortium</strong></p>
<p>The raw GWAS was downloaded from (<a href="https://www.med.unc.edu/pgc/download-results/scz/" class="uri">https://www.med.unc.edu/pgc/download-results/scz/</a>), EUR only descent. They are discussed in this paper: <a href="https://www.nature.com/articles/s41588-019-0512-x" class="uri">https://www.nature.com/articles/s41588-019-0512-x</a></p>
<p>First, we can plot the trait increasing allele frequency vs effect size for GWAS significant SNPs (P &lt; 5e-8), picking the lowest p-value per LD Block and coloring by derived status. (n=21)</p>
<pre class="r"><code>data &lt;- fread(&quot;~/scratch/SCZ-EAS_PGC.txt&quot;)
sig &lt;- sig_snps(data, 5e-8)
ld_block &lt;- lowest_pval(sig)
make_plot(ld_block, &quot;SCZ-EAS_PGC&quot;)</code></pre>
<p><img src="figure/misc_gwas.Rmd/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Here is the same plot with all the significant SNPs (not picking the lowest per LD Block).</p>
<pre class="r"><code>make_plot(sig, &quot;SCZ-EAS_PGC, all SNPs&quot;)</code></pre>
<p><img src="figure/misc_gwas.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-13-1">
Past versions of unnamed-chunk-13-1.png
</button>
</p>
<div id="fig-unnamed-chunk-13-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/jgblanc/infer_mutational_bias/blob/845a11ed8974fff944ca18efb108928918ad9a44/docs/figure/misc_gwas.Rmd/unnamed-chunk-13-1.png" target="_blank">845a11e</a>
</td>
<td>
jgblanc
</td>
<td>
2020-06-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Let’s try dropping the significance threshold (P &lt; 5e-4), still picking the lowest p-value per LD Block.</p>
<pre class="r"><code>sig &lt;- sig_snps(data, 5e-6)
make_plot(sig, &quot;SCZ-EAS_PGC&quot;)</code></pre>
<p><img src="figure/misc_gwas.Rmd/unnamed-chunk-14-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-1">
Past versions of unnamed-chunk-14-1.png
</button>
</p>
<div id="fig-unnamed-chunk-14-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/jgblanc/infer_mutational_bias/blob/845a11ed8974fff944ca18efb108928918ad9a44/docs/figure/misc_gwas.Rmd/unnamed-chunk-14-1.png" target="_blank">845a11e</a>
</td>
<td>
jgblanc
</td>
<td>
2020-06-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><strong>EUR Walters Group Data</strong></p>
<p>The raw GWAS was downloaded from (<a href="https://walters.psycm.cf.ac.uk/" class="uri">https://walters.psycm.cf.ac.uk/</a>, EUR only descent. They are discussed in this paper: <a href="https://www.nature.com/articles/s41588-018-0059-2" class="uri">https://www.nature.com/articles/s41588-018-0059-2</a></p>
<p>First, we can plot the trait increasing allele frequency vs effect size for GWAS significant SNPs (P &lt; 5e-8), picking the lowest p-value per LD Block and coloring by derived status.</p>
<pre class="r"><code>data &lt;- fread(&quot;~/scratch/SCZ_3982.txt&quot;)
sig &lt;- sig_snps(data, 5e-8)
ld_block &lt;- lowest_pval(sig)
make_plot(ld_block, &quot;SCZ-Walters&quot;)</code></pre>
<p><img src="figure/misc_gwas.Rmd/unnamed-chunk-15-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Now we can make the same plot but dropping all SNPs on chromosome 6.</p>
<pre class="r"><code>drop_six &lt;- ld_block %&gt;% filter(CHR != 6)
make_plot(drop_six, &quot;Exclude CHR 6&quot;)</code></pre>
<p><img src="figure/misc_gwas.Rmd/unnamed-chunk-16-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-16-1">
Past versions of unnamed-chunk-16-1.png
</button>
</p>
<div id="fig-unnamed-chunk-16-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/jgblanc/infer_mutational_bias/blob/845a11ed8974fff944ca18efb108928918ad9a44/docs/figure/misc_gwas.Rmd/unnamed-chunk-16-1.png" target="_blank">845a11e</a>
</td>
<td>
jgblanc
</td>
<td>
2020-06-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.2 (2019-12-12)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS High Sierra 10.13.6

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] ggpubr_0.3.0      knitr_1.29        forcats_0.5.0     stringr_1.4.0    
 [5] dplyr_1.0.0       purrr_0.3.4       readr_1.3.1       tidyr_1.1.0      
 [9] tibble_3.0.1      ggplot2_3.3.2     tidyverse_1.3.0   data.table_1.12.8
[13] workflowr_1.6.2  

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.4.6     lubridate_1.7.9  lattice_0.20-41  assertthat_0.2.1
 [5] rprojroot_1.3-2  digest_0.6.25    R6_2.4.1         cellranger_1.1.0
 [9] backports_1.1.8  reprex_0.3.0     evaluate_0.14    httr_1.4.1      
[13] pillar_1.4.4     rlang_0.4.6      curl_4.3         readxl_1.3.1    
[17] rstudioapi_0.11  whisker_0.4      car_3.0-8        blob_1.2.1      
[21] rmarkdown_2.3    labeling_0.3     foreign_0.8-72   munsell_0.5.0   
[25] broom_0.5.6      compiler_3.6.2   httpuv_1.5.4     modelr_0.1.8    
[29] xfun_0.15        pkgconfig_2.0.3  htmltools_0.5.0  tidyselect_1.1.0
[33] rio_0.5.16       fansi_0.4.1      crayon_1.3.4     dbplyr_1.4.4    
[37] withr_2.2.0      later_1.1.0.1    grid_3.6.2       nlme_3.1-148    
[41] jsonlite_1.6.1   gtable_0.3.0     lifecycle_0.2.0  DBI_1.1.0       
[45] git2r_0.27.1     magrittr_1.5     scales_1.1.1     zip_2.0.4       
[49] carData_3.0-4    cli_2.0.2        stringi_1.4.6    farver_2.0.3    
[53] ggsignif_0.6.0   fs_1.4.1         promises_1.1.1   xml2_1.3.2      
[57] ellipsis_0.3.1   generics_0.0.2   vctrs_0.3.1      cowplot_1.0.0   
[61] openxlsx_4.1.5   tools_3.6.2      glue_1.4.1       hms_0.5.3       
[65] abind_1.4-5      yaml_2.2.1       colorspace_1.4-1 rstatix_0.6.0   
[69] rvest_0.3.5      haven_2.3.1     </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
