<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Jennifer Blanc" />


<title>Stratification</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">infer_mutational_bias</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jgblanc/infer_mutational_bias">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Stratification</h1>
<h4 class="author">Jennifer Blanc</h4>
<h4 class="date">2/13/2020</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-02-26
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>infer_mutational_bias/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.4.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20191108code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20191108)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20191108code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20191108)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomjgblancinfermutationalbiastree9e721a39f5426ac5b7ee5f76fe2d12a5237e694ctargetblank9e721a3a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/jgblanc/infer_mutational_bias/tree/9e721a39f5426ac5b7ee5f76fe2d12a5237e694c" target="_blank">9e721a3</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomjgblancinfermutationalbiastree9e721a39f5426ac5b7ee5f76fe2d12a5237e694ctargetblank9e721a3a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    .snakemake/conda-archive/
    Ignored:    .snakemake/conda/
    Ignored:    .snakemake/locks/
    Ignored:    .snakemake/shadow/
    Ignored:    .snakemake/singularity/
    Ignored:    analysis/.DS_Store
    Ignored:    code/.DS_Store
    Ignored:    code/parse_gwas_atlas/.DS_Store
    Ignored:    data/.ipynb_checkpoints/
    Ignored:    data/47UKBB/
    Ignored:    data/BBJ/
    Ignored:    data/GWAS_ATLAS/
    Ignored:    data/STRAT/
    Ignored:    data/UKBB/
    Ignored:    docs/.DS_Store
    Ignored:    docs/assets/.DS_Store
    Ignored:    output/.DS_Store
    Ignored:    output/47UKBB/
    Ignored:    output/GWAS_ATLAS/

Untracked files:
    Untracked:  .snakemake/log/2020-01-30T145351.841085.snakemake.log
    Untracked:  .snakemake/log/2020-01-30T152809.242567.snakemake.log
    Untracked:  .snakemake/log/2020-01-30T152814.933120.snakemake.log
    Untracked:  .snakemake/log/2020-01-30T161043.605380.snakemake.log
    Untracked:  .snakemake/log/2020-01-30T163643.702329.snakemake.log
    Untracked:  .snakemake/log/2020-01-30T163914.317308.snakemake.log
    Untracked:  .snakemake/log/2020-01-30T165942.174064.snakemake.log
    Untracked:  .snakemake/log/2020-01-31T132509.011706.snakemake.log
    Untracked:  .snakemake/log/2020-01-31T132531.112131.snakemake.log
    Untracked:  .snakemake/log/2020-01-31T135909.720564.snakemake.log
    Untracked:  .snakemake/log/2020-01-31T135932.405589.snakemake.log
    Untracked:  .snakemake/log/2020-01-31T140258.528215.snakemake.log
    Untracked:  .snakemake/log/2020-01-31T154259.739904.snakemake.log
    Untracked:  .snakemake/log/2020-01-31T161214.457782.snakemake.log
    Untracked:  .snakemake/log/2020-02-04T122755.245485.snakemake.log
    Untracked:  .snakemake/log/2020-02-04T125925.054486.snakemake.log
    Untracked:  .snakemake/log/2020-02-04T131309.070607.snakemake.log
    Untracked:  .snakemake/log/2020-02-04T165741.607714.snakemake.log
    Untracked:  .snakemake/log/2020-02-04T165832.953657.snakemake.log
    Untracked:  .snakemake/log/2020-02-04T165845.720203.snakemake.log
    Untracked:  .snakemake/log/2020-02-04T170106.750876.snakemake.log
    Untracked:  .snakemake/log/2020-02-04T173554.946756.snakemake.log
    Untracked:  .snakemake/log/2020-02-04T203921.856975.snakemake.log
    Untracked:  .snakemake/log/2020-02-04T203936.733129.snakemake.log
    Untracked:  .snakemake/log/2020-02-05T112056.885301.snakemake.log
    Untracked:  .snakemake/log/2020-02-05T112242.653088.snakemake.log
    Untracked:  .snakemake/log/2020-02-05T112254.668205.snakemake.log
    Untracked:  .snakemake/log/2020-02-05T112317.962078.snakemake.log
    Untracked:  .snakemake/log/2020-02-05T114409.152072.snakemake.log
    Untracked:  .snakemake/log/2020-02-05T115017.456859.snakemake.log
    Untracked:  .snakemake/log/2020-02-05T115225.611708.snakemake.log
    Untracked:  .snakemake/log/2020-02-05T131627.566541.snakemake.log
    Untracked:  .snakemake/log/2020-02-05T135142.429095.snakemake.log
    Untracked:  .snakemake/log/2020-02-05T135147.725230.snakemake.log
    Untracked:  .snakemake/log/2020-02-06T170947.682904.snakemake.log
    Untracked:  .snakemake/log/2020-02-06T170955.880903.snakemake.log
    Untracked:  .snakemake/log/2020-02-06T171020.234309.snakemake.log
    Untracked:  .snakemake/log/2020-02-06T171059.838074.snakemake.log
    Untracked:  .snakemake/log/2020-02-06T171138.722326.snakemake.log
    Untracked:  .snakemake/log/2020-02-06T172326.673867.snakemake.log
    Untracked:  .snakemake/log/2020-02-09T161433.704127.snakemake.log
    Untracked:  .snakemake/log/2020-02-09T161518.690570.snakemake.log
    Untracked:  .snakemake/log/2020-02-09T161701.456606.snakemake.log
    Untracked:  .snakemake/log/2020-02-09T161908.570560.snakemake.log
    Untracked:  .snakemake/log/2020-02-09T163531.063766.snakemake.log
    Untracked:  .snakemake/log/2020-02-09T164707.068682.snakemake.log
    Untracked:  .snakemake/log/2020-02-09T164713.498823.snakemake.log
    Untracked:  .snakemake/log/2020-02-09T164933.856759.snakemake.log
    Untracked:  .snakemake/log/2020-02-09T165246.623702.snakemake.log
    Untracked:  .snakemake/log/2020-02-09T165729.736217.snakemake.log
    Untracked:  .snakemake/log/2020-02-09T165753.884876.snakemake.log
    Untracked:  .snakemake/log/2020-02-09T165811.808410.snakemake.log
    Untracked:  .snakemake/log/2020-02-09T170118.509824.snakemake.log
    Untracked:  .snakemake/log/2020-02-09T185943.187754.snakemake.log
    Untracked:  .snakemake/log/2020-02-09T201412.006222.snakemake.log
    Untracked:  .snakemake/log/2020-02-09T201442.388739.snakemake.log
    Untracked:  .snakemake/log/2020-02-09T203655.598430.snakemake.log
    Untracked:  .snakemake/log/2020-02-10T123520.594963.snakemake.log
    Untracked:  .snakemake/log/2020-02-10T140913.222118.snakemake.log
    Untracked:  .snakemake/log/2020-02-10T141224.194342.snakemake.log
    Untracked:  .snakemake/log/2020-02-10T142516.378779.snakemake.log
    Untracked:  .snakemake/log/2020-02-10T150055.924168.snakemake.log
    Untracked:  .snakemake/log/2020-02-10T150106.868447.snakemake.log
    Untracked:  .snakemake/metadata/b3V0cHV0L0JCSi9wYXJzZWRfZ3dhcy9CTUlfMV9wYXJzZWQudHh0
    Untracked:  .snakemake/metadata/b3V0cHV0L0JCSi9wYXJzZWRfZ3dhcy9IRExfNDdfcGFyc2VkLnR4dA==
    Untracked:  .snakemake/metadata/b3V0cHV0L0JCSi9wYXJzZWRfZ3dhcy9IYl80OV9wYXJzZWQudHh0
    Untracked:  .snakemake/metadata/b3V0cHV0L0JCSi9wYXJzZWRfZ3dhcy9IZWlnaHRfMTM1X3BhcnNlZC50eHQ=
    Untracked:  .snakemake/metadata/b3V0cHV0L0JCSi9wYXJzZWRfZ3dhcy9MRExfNjFfcGFyc2VkLnR4dA==
    Untracked:  .snakemake/metadata/b3V0cHV0L0JCSi9wYXJzZWRfZ3dhcy9QbHRfOTdfcGFyc2VkLnR4dA==
    Untracked:  .snakemake/metadata/b3V0cHV0L0JCSi9wYXJzZWRfZ3dhcy9UR18xMDlfcGFyc2VkLnR4dA==
    Untracked:  .snakemake/metadata/b3V0cHV0L0dXQVNfQVRMQVMvZXZvX2FkZGVkL01TXzI0MDc2NjAyX2V2by50eHQ=
    Untracked:  .snakemake/metadata/b3V0cHV0L0dXQVNfQVRMQVMvZXZvX2FkZGVkL0FEXzQwOTRfZXZvLnR4dA==
    Untracked:  .snakemake/metadata/b3V0cHV0L0dXQVNfQVRMQVMvZXZvX2FkZGVkL0FPYXN0aG1hXzQyNjdfZXZvLnR4dA==
    Untracked:  .snakemake/metadata/b3V0cHV0L0dXQVNfQVRMQVMvZXZvX2FkZGVkL0JNSV80MDc0X2V2by50eHQ=
    Untracked:  .snakemake/metadata/b3V0cHV0L0dXQVNfQVRMQVMvZXZvX2FkZGVkL0NPYXN0aG1hXzQyNjZfZXZvLnR4dA==
    Untracked:  .snakemake/metadata/b3V0cHV0L0dXQVNfQVRMQVMvZXZvX2FkZGVkL0VBbWV0YV80MDY2X2V2by50eHQ=
    Untracked:  .snakemake/metadata/b3V0cHV0L0dXQVNfQVRMQVMvZXZvX2FkZGVkL0VBdWtiYl8zNDA5X2V2by50eHQ=
    Untracked:  .snakemake/metadata/b3V0cHV0L0dXQVNfQVRMQVMvZXZvX2FkZGVkL0lCRF8yMDMxX2V2by50eHQ=
    Untracked:  .snakemake/metadata/b3V0cHV0L0dXQVNfQVRMQVMvZXZvX2FkZGVkL1VDXzIwMzBfZXZvLnR4dA==
    Untracked:  .snakemake/metadata/b3V0cHV0L0dXQVNfQVRMQVMvZXZvX2FkZGVkL215b3BpYV8zNTM5X2V2by50eHQ=
    Untracked:  .snakemake/metadata/b3V0cHV0L0dXQVNfQVRMQVMvZXZvX2FkZGVkL2JyZWFzdGNhbmNlcl8zNjU5X2V2by50eHQ=
    Untracked:  .snakemake/metadata/b3V0cHV0L0dXQVNfQVRMQVMvZXZvX2FkZGVkL3BhcmtpbnNvbl80MTY3X2V2by50eHQ=
    Untracked:  .snakemake/metadata/b3V0cHV0L0dXQVNfQVRMQVMvZXZvX2FkZGVkL3ZpdGlsaWdvXzM4MzJfZXZvLnR4dA==
    Untracked:  .snakemake/metadata/b3V0cHV0L0dXQVNfQVRMQVMvcGFyc2VkX2d3YXMvQ09hc3RobWFfNDI2Nl9wYXJzZWQudHh0
    Untracked:  .snakemake/metadata/b3V0cHV0L0dXQVNfQVRMQVMvcGFyc2VkX2d3YXMvQU9hc3RobWFfNDI2N19wYXJzZWQudHh0
    Untracked:  .snakemake/metadata/b3V0cHV0L0dXQVNfQVRMQVMvcGFyc2VkX2d3YXMvQURfNDA5NF9wYXJzZWQudHh0
    Untracked:  .snakemake/metadata/b3V0cHV0L0dXQVNfQVRMQVMvcGFyc2VkX2d3YXMvQk1JXzQwNzRfcGFyc2VkLnR4dA==
    Untracked:  .snakemake/metadata/b3V0cHV0L0dXQVNfQVRMQVMvcGFyc2VkX2d3YXMvRUF1a2JiXzM0MDlfcGFyc2VkLnR4dA==
    Untracked:  .snakemake/metadata/b3V0cHV0L0dXQVNfQVRMQVMvcGFyc2VkX2d3YXMvRUFtZXRhXzQwNjZfcGFyc2VkLnR4dA==
    Untracked:  .snakemake/metadata/b3V0cHV0L0dXQVNfQVRMQVMvcGFyc2VkX2d3YXMvSUJEXzIwMzFfcGFyc2VkLnR4dA==
    Untracked:  .snakemake/metadata/b3V0cHV0L0dXQVNfQVRMQVMvcGFyc2VkX2d3YXMvTVNfMjQwNzY2MDJfcGFyc2VkLnR4dA==
    Untracked:  .snakemake/metadata/b3V0cHV0L0dXQVNfQVRMQVMvcGFyc2VkX2d3YXMvVUNfMjAzMF9wYXJzZWQudHh0
    Untracked:  .snakemake/metadata/b3V0cHV0L0dXQVNfQVRMQVMvcGFyc2VkX2d3YXMvbXlvcGlhXzM1MzlfcGFyc2VkLnR4dA==
    Untracked:  .snakemake/metadata/b3V0cHV0L0dXQVNfQVRMQVMvcGFyc2VkX2d3YXMvcGFya2luc29uXzQxNjdfcGFyc2VkLnR4dA==
    Untracked:  .snakemake/metadata/b3V0cHV0L1VLQkIvcGFyc2VkX2d3YXMvQk1JXzIzMTA0X3BhcnNlZC50eHQ=
    Untracked:  .snakemake/metadata/b3V0cHV0L1VLQkIvcGFyc2VkX2d3YXMvSERMXzMwNzYwX3BhcnNlZC50eHQ=
    Untracked:  .snakemake/metadata/b3V0cHV0L1VLQkIvcGFyc2VkX2d3YXMvSGJfMzAwNTBfcGFyc2VkLnR4dA==
    Untracked:  .snakemake/metadata/b3V0cHV0L1VLQkIvcGFyc2VkX2d3YXMvSGVpZ2h0XzUwX3BhcnNlZC50eHQ=
    Untracked:  .snakemake/metadata/b3V0cHV0L1VLQkIvcGFyc2VkX2d3YXMvTERMXzMwNzgwX3BhcnNlZC50eHQ=
    Untracked:  .snakemake/metadata/b3V0cHV0L1VLQkIvcGFyc2VkX2d3YXMvUGx0XzMwMDgwX3BhcnNlZC50eHQ=
    Untracked:  .snakemake/metadata/b3V0cHV0L1VLQkIvcGFyc2VkX2d3YXMvVEdfMzA4NzBfcGFyc2VkLnR4dA==
    Untracked:  code/Neal_UKBB.R
    Untracked:  code/parse_BBJ/
    Untracked:  code/parse_gwas_atlas/AD_4094.R
    Untracked:  code/parse_gwas_atlas/AOasthma_4267.R
    Untracked:  code/parse_gwas_atlas/BMI_4074.R
    Untracked:  code/parse_gwas_atlas/COasthma_4266.R
    Untracked:  code/parse_gwas_atlas/EAmeta_4066.R
    Untracked:  code/parse_gwas_atlas/EAukbb_3409.R
    Untracked:  code/parse_gwas_atlas/EAukbb_3509.R
    Untracked:  code/parse_gwas_atlas/IBD_2031.R
    Untracked:  code/parse_gwas_atlas/MS_24076602.R
    Untracked:  code/parse_gwas_atlas/UC_2030.R
    Untracked:  code/parse_gwas_atlas/myopia_3539.R
    Untracked:  code/parse_gwas_atlas/myopic_3539.R
    Untracked:  code/parse_gwas_atlas/parkinson_4167.R
    Untracked:  output/BBJ/
    Untracked:  output/UKBB/

Unstaged changes:
    Modified:   .snakemake/metadata/b3V0cHV0L0dXQVNfQVRMQVMvcGFyc2VkX2d3YXMvYnJlYXN0Y2FuY2VyXzM2NTlfcGFyc2VkLnR4dA==
    Modified:   .snakemake/metadata/b3V0cHV0L0dXQVNfQVRMQVMvcGFyc2VkX2d3YXMvdml0aWxpZ29fMzgzMl9wYXJzZWQudHh0
    Deleted:    code/#get_evolutionary_information_from_1kg_GWAS_ATLAS.py#
    Modified:   code/.ipynb_checkpoints/Untitled-checkpoint.ipynb
    Modified:   code/Untitled.ipynb
    Modified:   code/get_evolutionary_information_from_1kg_GWAS_ATLAS.py
    Modified:   code/get_evolutionary_information_from_1kg_GWAS_ATLAS.py~
    Modified:   code/parse_gwas_atlas/vitiligo_3832.R
    Modified:   snakefile
    Modified:   snakefile~

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jgblanc/infer_mutational_bias/blob/9e721a39f5426ac5b7ee5f76fe2d12a5237e694c/analysis/Stratification.Rmd" target="_blank">9e721a3</a>
</td>
<td>
jgblanc
</td>
<td>
2020-02-26
</td>
<td>
Added PCA results
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>library(ggplot2)
library(data.table)
library(dplyr)
library(ggpubr)</code></pre>
<div id="intro" class="section level2">
<h2>Intro</h2>
<p>Here we are interested in detecting the correlation of derived alleles and individual level PCs. Recall that the singular value decomposition of the genotype matrix can be written as:</p>
<p><span class="math display">\[G_{n\times L} = U_{n \times n} \space \Lambda_{n \times n}^{1/2} \space V_{n \times L}\]</span></p>
<p>Here G is the genotype matrix. The columns of <span class="math inline">\(U\)</span> are the left Eigenvectors, the rows of of <span class="math inline">\(V\)</span> are the right eigenvectors and the diagonals of <span class="math inline">\(\Lambda^{1/2}\)</span> are the square root of the eigenvalues. We can connect this to the eigen decomposition of the genotype covariance matrix:</p>
<p><span class="math display">\[K_{n\times n} = GG^T = U \Lambda^{1/2}V \space G^T\]</span> <span class="math display">\[ = U \Lambda^{1/2}V \space V^T\Lambda^{1/2}U^T\]</span> <span class="math display">\[ = U \Lambda^{1/2} I \Lambda^{1/2}U^T\]</span> <span class="math display">\[ = U_{n \times n} \Lambda_{n \times n} U^T_{n \times n}\]</span></p>
<p>Here we are generating the square covariance matrix <span class="math inline">\(K\)</span> by multiplying <span class="math inline">\(G G^T\)</span>. Now we can substitute the SVD decomposition of <span class="math inline">\(G\)</span> from above. We can also transpose this SVD to get <span class="math inline">\(G^T\)</span>. Finally we can cancel things out until we get <span class="math inline">\(U \Lambda U^T\)</span>, the eigen decomposition. Here <span class="math inline">\(U\)</span> (the same as above) contains the eigenvectors in the columns and the diagonals have the eigen values (amount of variance explained by each eigenvector). An important thing to remember is that these eigenvectors describe the axes of variance for the individuals, not SNPs.</p>
<p>Similarly we can write the LD matrix, <span class="math inline">\(R\)</span> in terms of its SVD components:</p>
<p><span class="math display">\[R_{L\times L} = G^TG\]</span> <span class="math display">\[ = V^T \Lambda^{1/2} U^T \space U \Lambda^{1/2}V\]</span> <span class="math display">\[ = V_{L \times n}^T \Lambda_{n \times n}V_{n \times L}\]</span></p>
<p>Here we are interested in the right eigenvectors (the columns of <span class="math inline">\(V^T\)</span>) that represent the correlation of each SNP with the individual eigenvector. In other words, the covariance of SNP <span class="math inline">\(l\)</span> with the an individual eigenvector <span class="math inline">\(k\)</span> is given by the dot product of the eigenvector and the genotypes at the SNP:</p>
<p><span class="math display">\[\vec{U_k} \cdot \vec{g_l} = V_{k,l}\]</span></p>
<p>These <span class="math inline">\(V_{k,l}\)</span> entries represent the correlation of each SNP with the individual PC, also called the SNP “loadings”. Here we are interested if derived alleles are correlated with PC’s in the data set. For instance, the first PC in the 1kg EUR individuals corresponds geographically to the latitude. We are interested in the sum of the SNP loadings for this PC where the loading is positive if the coded allele is derived and negative if the coded allele is ancestral. This will tell us if there is a relationship between derived alleles and this PC (if there is no relationship then the sum is 0).</p>
</div>
<div id="method" class="section level2">
<h2>Method</h2>
<ol style="list-style-type: decimal">
<li>Took 1kg BCF files from the 4th floor comp bio data<br />
</li>
<li>Used plink to run PCA on only the EUR samples (n = 503)<br />
plink -bcf ../../../data/1kg/bcf/ALL.chr1.phase3_shapeit2_mvncall_integrated_v5.20130502.genotypes.bcf –keep ../../../data/1kg/bcf/EUR_IDS.txt –double-id –pca var-wts –out ../output/PCA/chr1_EUR –allow-extra-chr<br />
</li>
<li>Plink outputs the SNP “loadings” for the first 20 PC’s in an .eigenvec.var file<br />
</li>
<li>I used the 1kg_snps.tsv from Joe to add a column to the .eigenvec.var that says 1 if the coded allele (A2) is derived and -1 if the coded allele is ancestral. I was only able to annotate SNPs that had rsID’s and had derived/ancestral info in the table - the rest have NA in that column.</li>
</ol>
</div>
<div id="get-ids-of-european-individuals" class="section level2">
<h2>Get ID’s of European individuals</h2>
<pre class="r"><code>pop_info &lt;- fread(&quot;../data/STRAT/20130606_sample_info.csv&quot;)
EUR &lt;- pop_info %&gt;% filter(Population %in% c(&quot;CEU&quot;, &quot;TSI&quot;, &quot;FIN&quot;, &quot;GBR&quot;,&quot;IBS&quot;))
id_list &lt;- cbind(0,EUR$Sample)
id_only &lt;- cbind(EUR$Sample,EUR$Sample)
#write.table(id_only, &quot;../data/STRAT/EUR_IDS.txt&quot;, row.names = F, quote = F, col.names = F)</code></pre>
</div>
<div id="pca" class="section level2">
<h2>PCA</h2>
<pre class="r"><code>vecs &lt;- fread(&quot;../data/STRAT/chr1_EUR_0.01.eigenvec&quot;)
pop_info &lt;- fread(&quot;../data/STRAT/20130606_sample_info.csv&quot;)
dat &lt;- inner_join(vecs, pop_info, by = c(&quot;V2&quot;= &quot;Sample&quot;))
p1 &lt;- ggplot(data = dat, aes(x=V3, y=V4, color = Population)) + geom_point() + xlab(&quot;PC1&quot;) + ylab(&quot;PC2&quot;)
p2 &lt;- ggplot(data = dat, aes(x=V5, y=V6, color = Population)) + geom_point() + xlab(&quot;PC3&quot;) + ylab(&quot;PC4&quot;)
p3 &lt;- ggplot(data = dat, aes(x=V7, y=V8, color = Population)) + geom_point() + xlab(&quot;PC5&quot;) + ylab(&quot;PC6&quot;)
p4 &lt;- ggplot(data = dat, aes(x=V9, y=V10, color = Population)) + geom_point() + xlab(&quot;PC7&quot;) + ylab(&quot;PC8&quot;)
ggarrange(p1,p2,p3,p4, common.legend = T)</code></pre>
<p><img src="figure/Stratification.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="snp-loadings" class="section level2">
<h2>SNP Loadings</h2>
<pre class="r"><code>loading &lt;- fread(&quot;../data/STRAT/chr1_EUR_0.01_DA.txt&quot;)
loading &lt;- loading %&gt;% na.omit() # Get rid of no info (DA) SNPs 
only_num &lt;- loading[,5:24]
sums &lt;- colSums(only_num, na.rm = T)

df &lt;- as.data.frame(cbind(sums,seq(1,20)))
ggplot(df, aes(x=sums,y=V2)) + geom_point() + geom_vline(xintercept = 0, color = &quot;red&quot;) + ylab(&quot;PC&quot;) + scale_y_continuous(minor_breaks = seq(1 , 20, 1), breaks = seq(1, 20, 1))</code></pre>
<p><img src="figure/Stratification.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.5.1 (2018-07-02)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS High Sierra 10.13.6

Matrix products: default
BLAS: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] ggpubr_0.2        magrittr_1.5      dplyr_0.8.1       data.table_1.12.2
[5] ggplot2_3.2.1    

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.3       compiler_3.5.1   pillar_1.4.2     git2r_0.25.2    
 [5] workflowr_1.4.0  tools_3.5.1      digest_0.6.22    evaluate_0.14   
 [9] lifecycle_0.1.0  tibble_2.1.3     gtable_0.3.0     pkgconfig_2.0.3 
[13] rlang_0.4.1      yaml_2.2.0       xfun_0.7         gridExtra_2.3   
[17] withr_2.1.2      stringr_1.4.0    knitr_1.23       fs_1.3.1        
[21] cowplot_0.9.4    rprojroot_1.3-2  grid_3.5.1       tidyselect_0.2.5
[25] glue_1.3.1       R6_2.4.1         rmarkdown_1.13   farver_2.0.1    
[29] purrr_0.3.2      whisker_0.3-2    backports_1.1.5  scales_1.1.0    
[33] htmltools_0.3.6  assertthat_0.2.1 colorspace_1.4-1 labeling_0.3    
[37] stringi_1.4.3    lazyeval_0.2.2   munsell_0.5.0    crayon_1.3.4    </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
