---
title: "Comparisons"
author: "Jennifer Blanc"
date: "2/10/2020"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(data.table)
library(ggplot2)
library(dplyr)
library(ggpubr)
library(latex2exp)
library(Hmisc)
```

## Intro 

I downloaded all the BBJ summary statistics from (http://jenger.riken.jp/en/result). The ID number of the GWAS comes after the trait name in the file name. The UKBB summary statistics cam from the Neale lab (https://docs.google.com/spreadsheets/d/1kvPoupSzsSFBNSztMzl04xMoSC3Kcx3CrjVf4yBmESU/edit?ts=5b5f17db#gid=227859291). For height and BMI I used the "raw" files that hadn't been transformed, for all the other traits I used the "irnt" files that used a inverse rank normal transformation. I used my own scripts to standardize the files and calculate the the effect allele frequecy. Here I loaded both the files for both biobanks and polorize effect size to be the BETA for the minor allele. I then binned the SNPs based on MAF to 100 bins with approximately equal numbers of SNPs. I then calculated the the average effect size per bin. Finally I plot the frequency (the max frequency of the bin) vs the mean effect size per bin. The main thing to note is that I haven't LD clumped any of these summary statistics yet...I'm working on setting up an LD clumping pipeline using PLINK but haven't finished yet.   

## Tratis 

```{r}
get_ES <- function(df) {
  df$ES <- NA
  minor_ef <- subset(df,df$EAF == df$MAF)
  minor_ef$ES <- minor_ef$BETA
  major_ef <- subset(df, df$EAF != df$MAF)
  major_ef$ES <- -1 * major_ef$BETA
  return(rbind(minor_ef,major_ef))
} 

bin_data <- function(df) {
  df <- df %>% select(ES,MAF)
  df <- df[order(MAF), ]
  df$BIN <- as.numeric(cut2(df$MAF, g=100))
  return(df)
}

group_dat <- function(df) {
  dat <- df %>% group_by(BIN) %>% summarise(AVG = mean(ES), MAX = max(MAF), NUM_SNPS = n(), SE = sd(ES))
}

#geom_pointrange(aes(ymin=AVG-SE, ymax=AVG+SE))
```


**Height**  
```{r, warning=FALSE}
uk_height <- group_dat(bin_data(get_ES(fread("../output/UKBB/parsed_gwas/Height_50_parsed.txt"))))
jb_height <- group_dat(bin_data(get_ES(fread("../output/BBJ/parsed_gwas/Height_135_parsed.txt"))))

pl1 <- ggplot(uk_height, aes(x=MAX, y=AVG)) + geom_point(color = "darkviolet", alpha = 0.5, size = 1.5) + theme_bw() + xlab("Frequency") + ylab("Mean Effect Size") + ggtitle("UKBB") + theme(plot.title = element_text(hjust = 0.5)) + geom_hline(yintercept = 0, color = "darkred", size = 1) + geom_smooth(method = "loess", size = 0.75, color = "navy") + lims(y=c(-3,0.1))
pl2 <- ggplot(jb_height, aes(x=MAX, y=AVG)) + geom_point(color= "darkgreen") + theme_bw()+ xlab("Frequency") + ylab("Mean Effect Size") + ggtitle("BBJ") + theme(plot.title = element_text(hjust = 0.5)) + geom_hline(yintercept = 0, color = "darkred", size = 1) + geom_smooth(method = "loess", size = 0.75, color = "navy") + lims(y=c(-3,0.1))

ggarrange(pl1, pl2)
```


Remove the lowest frequnecy bin 
```{r, warning=FALSE}
pl1 <- ggplot(uk_height[-1,], aes(x=MAX, y=AVG)) + geom_point(color = "darkviolet") + theme_bw() + xlab("Frequency") + ylab("Mean Effect Size") + ggtitle("UKBB") + theme(plot.title = element_text(hjust = 0.5)) + geom_hline(yintercept = 0, color = "darkred", size = 1) + geom_smooth(method = "loess", size = 0.75, color = "navy") + lims(y=c(-0.1,0.175))
pl2 <- ggplot(jb_height[-1,], aes(x=MAX, y=AVG)) + geom_point(color= "darkgreen") + theme_bw()+ xlab("Frequency") + ylab("Mean Effect Size") + ggtitle("BBJ") + theme(plot.title = element_text(hjust = 0.5)) + geom_hline(yintercept = 0, color = "darkred", size = 1) + geom_smooth(method = "loess", size = 0.75, color = "navy") + lims(y=c(-0.1, 0.175))

ggarrange(pl1, pl2)
```



**BMI** 
```{r}
uk_BMI <- group_dat(bin_data(get_ES(fread("../output/UKBB/parsed_gwas/BMI_23104_parsed.txt"))))
jb_BMI <- group_dat(bin_data(get_ES(fread("../output/BBJ/parsed_gwas/BMI_1_parsed.txt"))))

pl1 <- ggplot(uk_BMI, aes(x=MAX, y=AVG)) + geom_point(color = "darkviolet") + theme_bw() + xlab("Frequency") + ylab("Mean Effect Size") + ggtitle("UKBB") + theme(plot.title = element_text(hjust = 0.5)) + geom_hline(yintercept = 0, color = "darkred", size = 1) + geom_smooth(method = "loess", size = 0.75, color = "navy") + lims(y=c(-0.1,0.6))
pl2 <- ggplot(jb_BMI, aes(x=MAX, y=AVG)) + geom_point(color= "darkgreen") + theme_bw()+ xlab("Frequency") + ylab("Mean Effect Size") + ggtitle("BBJ") + theme(plot.title = element_text(hjust = 0.5)) + geom_hline(yintercept = 0, color = "darkred", size = 1) + geom_smooth(method = "loess", size = 0.75, color = "navy") + lims(y=c(-0.1,0.6))

ggarrange(pl1, pl2)
```

Remove lowest frequency bin 
```{r}
pl1 <- ggplot(uk_BMI[-1,], aes(x=MAX, y=AVG)) + geom_point(color = "darkviolet") + theme_bw() + xlab("Frequency") + ylab("Mean Effect Size") + ggtitle("UKBB") + theme(plot.title = element_text(hjust = 0.5)) + geom_hline(yintercept = 0, color = "darkred", size = 1) + geom_smooth(method = "loess", size = 0.75, color = "navy") + lims(y=c(-0.1,0.1))
pl2 <- ggplot(jb_BMI[-1,], aes(x=MAX, y=AVG)) + geom_point(color= "darkgreen") + theme_bw()+ xlab("Frequency") + ylab("Mean Effect Size") + ggtitle("BBJ") + theme(plot.title = element_text(hjust = 0.5)) + geom_hline(yintercept = 0, color = "darkred", size = 1) + geom_smooth(method = "loess", size = 0.75, color = "navy") + lims(y=c(-.1, 0.1))

ggarrange(pl1, pl2)
```



**HDL** 
```{r}
uk_HDL <- group_dat(bin_data(get_ES(fread("../output/UKBB/parsed_gwas/HDL_30760_parsed.txt"))))
jb_HDL <- group_dat(bin_data(get_ES(fread("../output/BBJ/parsed_gwas/HDL_47_parsed.txt"))))

pl1 <- ggplot(uk_HDL, aes(x=MAX, y=AVG)) + geom_point(color = "darkviolet") + theme_bw() + xlab("Frequency") + ylab("Mean Effect Size") + ggtitle("UKBB") + theme(plot.title = element_text(hjust = 0.5)) + geom_hline(yintercept = 0, color = "darkred", size = 1) + geom_smooth(method = "loess", size = 0.75, color = "navy") + lims(y=c(-0.1,0.12))
pl2 <- ggplot(jb_HDL, aes(x=MAX, y=AVG)) + geom_point(color= "darkgreen") + theme_bw()+ xlab("Frequency") + ylab("Mean Effect Size") + ggtitle("BBJ") + theme(plot.title = element_text(hjust = 0.5)) + geom_hline(yintercept = 0, color = "darkred", size = 1) + geom_smooth(method = "loess", size = 0.75, color = "navy") + lims(y=c(-0.1,0.12))

ggarrange(pl1, pl2)
```

Remove lowest frequency bin 
```{r}
pl1 <- ggplot(uk_HDL[-1,], aes(x=MAX, y=AVG)) + geom_point(color = "darkviolet") + theme_bw() + xlab("Frequency") + ylab("Mean Effect Size") + ggtitle("UKBB") + theme(plot.title = element_text(hjust = 0.5)) + geom_hline(yintercept = 0, color = "darkred", size = 1) + geom_smooth(method = "loess", size = 0.75, color = "navy") + lims(y=c(-0.1,0.12))
pl2 <- ggplot(jb_HDL[-1,], aes(x=MAX, y=AVG)) + geom_point(color= "darkgreen") + theme_bw()+ xlab("Frequency") + ylab("Mean Effect Size") + ggtitle("BBJ") + theme(plot.title = element_text(hjust = 0.5)) + geom_hline(yintercept = 0, color = "darkred", size = 1) + geom_smooth(method = "loess", size = 0.75, color = "navy") + lims(y=c(-0.1,0.12))

ggarrange(pl1, pl2)
```


**LDL** 
```{r}
uk_LDL <- group_dat(bin_data(get_ES(fread("../output/UKBB/parsed_gwas/LDL_30780_parsed.txt"))))
jb_LDL <- group_dat(bin_data(get_ES(fread("../output/BBJ/parsed_gwas/LDL_61_parsed.txt"))))

pl1 <- ggplot(uk_LDL, aes(x=MAX, y=AVG)) + geom_point(color = "darkviolet") + theme_bw() + xlab("Frequency") + ylab("Mean Effect Size") + ggtitle("UKBB") + theme(plot.title = element_text(hjust = 0.5)) + geom_hline(yintercept = 0, color = "darkred", size = 1) + geom_smooth(method = "loess", size = 0.75, color = "navy") + lims(y=c(-0.2,0.1))
pl2 <- ggplot(jb_LDL, aes(x=MAX, y=AVG)) + geom_point(color= "darkgreen") + theme_bw()+ xlab("Frequency") + ylab("Mean Effect Size") + ggtitle("BBJ") + theme(plot.title = element_text(hjust = 0.5)) + geom_hline(yintercept = 0, color = "darkred", size = 1) + geom_smooth(method = "loess", size = 0.75, color = "navy") + lims(y=c(-0.2,0.1))

ggarrange(pl1, pl2)
```

Remove lowest frequency bin 
```{r}
pl1 <- ggplot(uk_LDL[-1,], aes(x=MAX, y=AVG)) + geom_point(color = "darkviolet") + theme_bw() + xlab("Frequency") + ylab("Mean Effect Size") + ggtitle("UKBB") + theme(plot.title = element_text(hjust = 0.5)) + geom_hline(yintercept = 0, color = "darkred", size = 1) + geom_smooth(method = "loess", size = 0.75, color = "navy") + lims(y=c(-0.2,0.075))
pl2 <- ggplot(jb_LDL[-1,], aes(x=MAX, y=AVG)) + geom_point(color= "darkgreen") + theme_bw()+ xlab("Frequency") + ylab("Mean Effect Size") + ggtitle("BBJ") + theme(plot.title = element_text(hjust = 0.5)) + geom_hline(yintercept = 0, color = "darkred", size = 1) + geom_smooth(method = "loess", size = 0.75, color = "navy") + lims(y=c(-0.2,0.075))

ggarrange(pl1, pl2)
```


**Triglycerides** 
```{r}
uk_TG <- group_dat(bin_data(get_ES(fread("../output/UKBB/parsed_gwas/TG_30870_parsed.txt"))))
jb_TG <- group_dat(bin_data(get_ES(fread("../output/BBJ/parsed_gwas/TG_109_parsed.txt"))))

pl1 <- ggplot(uk_TG, aes(x=MAX, y=AVG)) + geom_point(color = "darkviolet") + theme_bw() + xlab("Frequency") + ylab("Mean Effect Size") + ggtitle("UKBB") + theme(plot.title = element_text(hjust = 0.5)) + geom_hline(yintercept = 0, color = "darkred", size = 1) + geom_smooth(method = "loess", size = 0.75, color = "navy") + lims(y=c(-0.15,0.1))
pl2 <- ggplot(jb_TG, aes(x=MAX, y=AVG)) + geom_point(color= "darkgreen") + theme_bw()+ xlab("Frequency") + ylab("Mean Effect Size") + ggtitle("BBJ") + theme(plot.title = element_text(hjust = 0.5)) + geom_hline(yintercept = 0, color = "darkred", size = 1) + geom_smooth(method = "loess", size = 0.75, color = "navy") + lims(y=c(-0.15,0.1))

ggarrange(pl1, pl2)
```

Remove lowest frequency bin 
```{r}
pl1 <- ggplot(uk_TG[-1,], aes(x=MAX, y=AVG)) + geom_point(color = "darkviolet") + theme_bw() + xlab("Frequency") + ylab("Mean Effect Size") + ggtitle("UKBB") + theme(plot.title = element_text(hjust = 0.5)) + geom_hline(yintercept = 0, color = "darkred", size = 1) + geom_smooth(method = "loess", size = 0.75, color = "navy") + lims(y=c(-0.15,0.075))
pl2 <- ggplot(jb_TG[-1,], aes(x=MAX, y=AVG)) + geom_point(color= "darkgreen") + theme_bw()+ xlab("Frequency") + ylab("Mean Effect Size") + ggtitle("BBJ") + theme(plot.title = element_text(hjust = 0.5)) + geom_hline(yintercept = 0, color = "darkred", size = 1) + geom_smooth(method = "loess", size = 0.75, color = "navy") + lims(y=c(-0.15,0.075))

ggarrange(pl1, pl2)
```

**Hemoglobin** 
```{r}
uk_Hb <- group_dat(bin_data(get_ES(fread("../output/UKBB/parsed_gwas/Hb_30050_parsed.txt"))))
jb_Hb <- group_dat(bin_data(get_ES(fread("../output/BBJ/parsed_gwas/Hb_49_parsed.txt"))))

pl1 <- ggplot(uk_Hb, aes(x=MAX, y=AVG)) + geom_point(color = "darkviolet") + theme_bw() + xlab("Frequency") + ylab("Mean Effect Size") + ggtitle("UKBB") + theme(plot.title = element_text(hjust = 0.5)) + geom_hline(yintercept = 0, color = "darkred", size = 1) + geom_smooth(method = "loess", size = 0.75, color = "navy") + lims(y=c(-0.1,0.07))
pl2 <- ggplot(jb_Hb, aes(x=MAX, y=AVG)) + geom_point(color= "darkgreen") + theme_bw()+ xlab("Frequency") + ylab("Mean Effect Size") + ggtitle("BBJ") + theme(plot.title = element_text(hjust = 0.5)) + geom_hline(yintercept = 0, color = "darkred", size = 1) + geom_smooth(method = "loess", size = 0.75, color = "navy") + lims(y=c(-0.1,0.07))

ggarrange(pl1, pl2)
```

Remove lowest frequency bin  
```{r}
pl1 <- ggplot(uk_Hb[-1,], aes(x=MAX, y=AVG)) + geom_point(color = "darkviolet") + theme_bw() + xlab("Frequency") + ylab("Mean Effect Size") + ggtitle("UKBB") + theme(plot.title = element_text(hjust = 0.5)) + geom_hline(yintercept = 0, color = "darkred", size = 1) + geom_smooth(method = "loess", size = 0.75, color = "navy") + lims(y=c(-0.1,0.07))
pl2 <- ggplot(jb_Hb[-1,], aes(x=MAX, y=AVG)) + geom_point(color= "darkgreen") + theme_bw()+ xlab("Frequency") + ylab("Mean Effect Size") + ggtitle("BBJ") + theme(plot.title = element_text(hjust = 0.5)) + geom_hline(yintercept = 0, color = "darkred", size = 1) + geom_smooth(method = "loess", size = 0.75, color = "navy") + lims(y=c(-0.1,0.07))

ggarrange(pl1, pl2)
```

**Platelet Count**  
```{r}
uk_Plt <- group_dat(bin_data(get_ES(fread("../output/UKBB/parsed_gwas/Plt_30080_parsed.txt"))))
jb_Plt <- group_dat(bin_data(get_ES(fread("../output/BBJ/parsed_gwas/Plt_97_parsed.txt"))))

pl1 <- ggplot(uk_Plt, aes(x=MAX, y=AVG)) + geom_point(color = "darkviolet") + theme_bw() + xlab("Frequency") + ylab("Mean Effect Size") + ggtitle("UKBB") + theme(plot.title = element_text(hjust = 0.5)) + geom_hline(yintercept = 0, color = "darkred", size = 1) + geom_smooth(method = "loess", size = 0.75, color = "navy") + lims(y=c(-0.07,0.05))
pl2 <- ggplot(jb_Plt, aes(x=MAX, y=AVG)) + geom_point(color= "darkgreen") + theme_bw()+ xlab("Frequency") + ylab("Mean Effect Size") + ggtitle("BBJ") + theme(plot.title = element_text(hjust = 0.5)) + geom_hline(yintercept = 0, color = "darkred", size = 1) + geom_smooth(method = "loess", size = 0.75, color = "navy") + lims(y=c(-0.07,0.05))

ggarrange(pl1, pl2)
```

Remove lowest frequency bin 
```{r}
pl1 <- ggplot(uk_Plt[-1,], aes(x=MAX, y=AVG)) + geom_point(color = "darkviolet") + theme_bw() + xlab("Frequency") + ylab("Mean Effect Size") + ggtitle("UKBB") + theme(plot.title = element_text(hjust = 0.5)) + geom_hline(yintercept = 0, color = "darkred", size = 1) + geom_smooth(method = "loess", size = 0.75, color = "navy") + lims(y=c(-0.07,0.05))
pl2 <- ggplot(jb_Plt[-1,], aes(x=MAX, y=AVG)) + geom_point(color= "darkgreen") + theme_bw()+ xlab("Frequency") + ylab("Mean Effect Size") + ggtitle("BBJ") + theme(plot.title = element_text(hjust = 0.5)) + geom_hline(yintercept = 0, color = "darkred", size = 1) + geom_smooth(method = "loess", size = 0.75, color = "navy") + lims(y=c(-0.07,0.05))

ggarrange(pl1, pl2)
```






