---
title: "GWAS_ATLAS_results"
author: "Jennifer Blanc"
date: "1/23/2020"
output: html_document
---

```{r, warning=FALSE, message=FALSE}
library(data.table)
library(ggplot2)
library(ggpubr)
library(latex2exp)
library(dplyr)
options(knitr.graphics.error = FALSE)
```


# Intro 

Write overview 

# Results 

```{r}
'%!in%' <- function(x,y)!('%in%'(x,y))
HLA <- seq(28477797,33448354)
remove_HLA <- function(df) {
  bad <- subset(df, df$CHR == 6 & df$BP %in% HLA)
  bad_snps <- bad$SNP
  good <- df %>% filter(SNP  %!in% bad_snps)
  return(good)
}
```


### Schizophrenia - Meta analysis  


Manhattan Plot  
```{r, out.height="75%", out.width="75%", echo=FALSE}
knitr::include_graphics("assets/schizophrenia_3982.png", error = FALSE)
```

RAF vs Effect Size 
```{r, warning=FALSE, echo=FALSE, message=FALSE}
cols <- c("deeppink4", "navy")
derive <- c("springgreen4", "purple1")
schizophrenia <- fread("../output/GWAS_ATLAS/evo_added/schizophrenia_3982_evo.txt")
colnames(schizophrenia) <- c(colnames(schizophrenia)[1:13], "EA_Derived", "DAF", "Risk_Derived")
schizophrenia$disease <- "Schizophenria"
clumped <- fread("../output/GWAS_ATLAS/clumped/schizophrenia_3982_SNPs.txt")
schizophrenia <- left_join(clumped, schizophrenia) 

#p1 <- ggplot(data=schizophrenia, aes(x=RAF,y=OR,color=Risk_Derived)) + geom_point(alpha=0.5) + theme_bw() + scale_color_manual(values = cols) + theme(plot.title = element_text(hjust = 0.5)) + ggtitle("OR")

#dat <- subset(schizophrenia, schizophrenia$Risk_Derived == 'T' | schizophrenia$Risk_Derived == 'F')
#p2 <- ggplot(data=dat, aes(x=RAF,y=abs(log(OR)),color=Risk_Derived)) + geom_point(alpha=0.5) + theme_bw() + scale_color_manual(values = derive, name = "Risk Allele Derived", labels = c("False", "True")) + theme(plot.title = element_text(hjust = 0.5)) + ylab(TeX('$\\hat{\\beta}$')) + xlab("Risk Alllele Frequency (RAF)") + theme(axis.text.x = element_text(size = 11), axis.text.y = element_text(size = 11), axis.title.y = element_text(size = 15))

#ggarrange(p1,p2, nrow = 2)

schizophrenia <- subset(schizophrenia, schizophrenia$Risk_Derived == 'T' | schizophrenia$Risk_Derived == 'F')
pl <- ggplot(data=schizophrenia, aes(x=RAF,y=abs(log(OR)),color=Risk_Derived)) + geom_point(alpha=0.5) + theme_bw() + scale_color_manual(values = derive) + theme(plot.title = element_text(hjust = 0.5),  legend.position = "bottom") + ggtitle("Schizophrenia") + labs(y = TeX('$\\hat{\\beta}$'), color = "Risk Derived") + ylim(0,max(abs(log(schizophrenia$OR)))+0.01)
pl
#ggsave("~/Desktop/schizophrenia.png",pl, height = 5, width = 9)

pl_nocolor <- ggplot(data=schizophrenia, aes(x=RAF,y=abs(log(OR)))) + geom_point(alpha=0.5) + theme_bw()  + theme(plot.title = element_text(hjust = 0.5),  legend.position = "bottom") + ggtitle("Schizophrenia") + labs(y = TeX('$\\hat{\\beta}$'), color = "Risk Derived") + ylim(0,max(abs(log(schizophrenia$OR)))+0.01)
pl_nocolor
```

Mean/Median RAF   
```{r, warning=F,echo=FALSE}
mean(schizophrenia$RAF)
median(schizophrenia$RAF)
```

Remove HLA Region 
```{r, warning=FALSE, echo=FALSE, message=FALSE}
dat <- remove_HLA(schizophrenia)
ggplot(data=dat, aes(x=RAF,y=abs(log(OR)),color=Risk_Derived)) + geom_point(alpha=0.5) + theme_bw() + scale_color_manual(values = derive) + theme(plot.title = element_text(hjust = 0.5),  legend.position = "bottom") + ggtitle("Schizophrenia") + labs(y = TeX('$\\hat{\\beta}$'), color = "Risk Derived") + ylim(0,max(abs(log(dat$OR)))+0.01)

ggplot(data=dat, aes(x=RAF,y=abs(log(OR)))) + geom_point(alpha=0.5) + theme_bw()  + theme(plot.title = element_text(hjust = 0.5),  legend.position = "bottom") + ggtitle("Schizophrenia") + labs(y = TeX('$\\hat{\\beta}$'), color = "Risk Derived") + ylim(0,max(abs(log(dat$OR)))+0.01)
```

Mean/Median RAF   
```{r, warning=F,echo=FALSE}
mean(dat$RAF)
median(dat$RAF)
```

Notes on SNP research  

### Schizophrenia - Psychiatric Genomic Consortium  

Manhattan Plot  
```{r, out.height="100%", out.width="100%", echo=FALSE}
knitr::include_graphics("assets/manhattan_atlas_4038.png", error = FALSE)
```

RAF vs Effect Size
```{r, warning=FALSE,echo=FALSE}
cols <- c("deeppink4", "navy")
derive <- c("springgreen4", "purple1")
scz_pgc <- fread("../output/GWAS_ATLAS/evo_added/schizophrenia_PGC_evo.txt")
colnames(scz_pgc) <- c(colnames(scz_pgc)[1:13], "EA_Derived", "DAF", "Risk_Derived")
clumped <- fread("../output/GWAS_ATLAS/clumped/schizophrenia_PGC_SNPs.txt")
scz_pgc <- left_join(clumped, scz_pgc, by= "SNP") 
scz_pgc$disease <- "Schizophenria PGC"

scz_pgc <- subset(scz_pgc, scz_pgc$Risk_Derived == 'T' | scz_pgc$Risk_Derived == 'F')
ggplot(data=scz_pgc, aes(x=RAF,y=abs(log(OR)),color=Risk_Derived)) + geom_point(alpha=0.5) + theme_bw() + scale_color_manual(values = derive) + theme(plot.title = element_text(hjust = 0.5)) + ggtitle("Schizophrenia PGC") + labs(y = TeX('$\\hat{\\beta}$'), color = "Risk Derived") + ylim(0,max(abs(log(scz_pgc$OR)))+0.01) + theme(plot.title = element_text(hjust = 0.5),  legend.position = "bottom")
```

Mean/Median RAF  
```{r, warning=F,echo=FALSE}
mean(scz_pgc$RAF)
median(scz_pgc$RAF)
```

Remove HLA Region
```{r, warning=F,echo=FALSE}
dat <- remove_HLA(scz_pgc)
ggplot(data=dat, aes(x=RAF,y=abs(log(OR)),color=Risk_Derived)) + geom_point(alpha=0.5) + theme_bw() + scale_color_manual(values = derive) + theme(plot.title = element_text(hjust = 0.5)) + ggtitle("Schizophrenia PGC") + labs(y = TeX('$\\hat{\\beta}$'), color = "Risk Derived") + ylim(0,max(abs(log(dat$OR)))+0.01) + theme(plot.title = element_text(hjust = 0.5),  legend.position = "bottom")
```
Mean/Median RAF  
```{r, warning=F,echo=FALSE}
mean(dat$RAF)
median(dat$RAF)
```



### Schizophrenia EAS - not LD clumped!  

Manhattan Plot  
```{r, out.height="100%", out.width="100%", echo=FALSE}
knitr::include_graphics("assets/scz_EAS.png", error = FALSE)
```

RAF vs Effect Size - not LD clumped
```{r, warning=FALSE, echo=FALSE}
scz_eas <- fread("../output/GWAS_ATLAS/evo_added/schizophrenia_EAS_evo.txt")
colnames(scz_eas) <- c(colnames(scz_eas)[1:13], "EA_Derived", "DAF", "Risk_Derived")
#clumped <- fread("../output/GWAS_ATLAS/clumped/schizophrenia_EAS.clumped")
#scz_eas <- left_join(clumped, scz_pgc, by= "SNP") 
scz_eas$disease <- "Schizophrenia EAS"

scz_eas <- subset(scz_eas, scz_eas$Risk_Derived == 'T' | scz_eas$Risk_Derived == 'F')
ggplot(data=scz_eas, aes(x=RAF,y=abs(log(OR)),color=Risk_Derived)) + geom_point(alpha=0.5) + theme_bw() + scale_color_manual(values = derive) + theme(plot.title = element_text(hjust = 0.5)) + ggtitle("Schizophrenia EAS") + labs(y = TeX('$\\hat{\\beta}$'), color = "Risk Derived") + ylim(0,max(abs(log(scz_eas$OR)))+0.01) + theme(plot.title = element_text(hjust = 0.5),  legend.position = "bottom")
#mean(scz_eas$RAF)
```

Remove HLA Region
```{r, warning=F,echo=FALSE}
dat <- remove_HLA(scz_eas)
ggplot(data=dat, aes(x=RAF,y=abs(log(OR)),color=Risk_Derived)) + geom_point(alpha=0.5) + theme_bw() + scale_color_manual(values = derive) + theme(plot.title = element_text(hjust = 0.5)) + ggtitle("Schizophrenia EAS") + labs(y = TeX('$\\hat{\\beta}$'), color = "Risk Derived") + ylim(0,max(abs(log(dat$OR)))+0.01) + theme(plot.title = element_text(hjust = 0.5),  legend.position = "bottom")
```

Mean/Median RAF  
```{r, warning=F,echo=FALSE}
mean(scz_eas$RAF)
median(scz_eas$RAF)
```


### Chronic Kidney Disease

Manhattan Plot  
```{r, out.height="100%", out.width="100%", echo=FALSE}
knitr::include_graphics("assets/manhattan_atlas_4210.png", error = FALSE)
```

RAF vs Effect Size 
```{r, warning=FALSE,echo=FALSE}
CKD <- fread("../output/GWAS_ATLAS/evo_added/CKD_4210_evo.txt")
colnames(CKD) <- c(colnames(CKD)[1:13], "EA_Derived", "DAF", "Risk_Derived")
CKD$disease <- "Chronic Kidney Disease"
clumped <- fread("../output/GWAS_ATLAS/clumped/CKD_4210.clumped")
CKD <- left_join(clumped, CKD, by= "SNP") 

ggplot(data=CKD, aes(x=RAF,y=abs(BETA),color=Risk_Derived)) + geom_point(alpha=0.5) + theme_bw() + scale_color_manual(values = derive) + theme(plot.title = element_text(hjust = 0.5)) + ggtitle("Risk Beta, colored by D/A") + ggtitle("Chronic Kidney Disease") + labs(y = TeX('$\\hat{\\beta}$'), color = "Risk Derived") + ylim(0,max(abs(CKD$BETA))+0.01) + theme(plot.title = element_text(hjust = 0.5),  legend.position = "bottom")
```

No HLA SNPs 

Mean/Median RAF  
```{r, warning=F,echo=FALSE}
mean(CKD$RAF)
median(CKD$RAF)
```

### Atrial Fibrilation  

Manhattan Plot 
```{r, out.height="100%", out.width="100%", echo=FALSE}
knitr::include_graphics("assets/manhattan_atlas_4361.png", error = FALSE)
```

RAF vs Effect Size  
```{r, warning=FALSE, echo=FALSE}
AF <- fread("../output/GWAS_ATLAS/evo_added/AF_4361_evo.txt")
colnames(AF) <- c(colnames(AF)[1:13], "EA_Derived", "DAF", "Risk_Derived")
AF$disease <- "Atrial Fibrilation"
clumped <- fread("../output/GWAS_ATLAS/clumped/AF_4361_SNPs.txt")
AF <- left_join(clumped, AF, by= "SNP") 

AF <- subset(AF, AF$Risk_Derived == 'T' | AF$Risk_Derived == 'F')
ggplot(data=AF, aes(x=RAF,y=abs(EFFECT),color=Risk_Derived)) + geom_point(alpha=0.5) + theme_bw() + scale_color_manual(values = derive) + theme(plot.title = element_text(hjust = 0.5)) + ggtitle("Atrial fibrillation") + labs(y = TeX('$\\hat{\\beta}$'), color = "Risk Derived") + ylim(0,max(abs(AF$EFFECT))+0.01) + theme(plot.title = element_text(hjust = 0.5),  legend.position = "bottom")
```

Atrial Fibrilation - remove top hit
```{r, warning=FALSE, echo=FALSE}
# Remove top hit 
AF <- subset(AF, AF$Risk_Derived == 'T' | AF$Risk_Derived == 'F')
AF_ind <- which.max(abs(AF$EFFECT)) 
AF <- AF[-487, ]
ggplot(data=AF, aes(x=RAF,y=abs(EFFECT),color=Risk_Derived)) + geom_point(alpha=0.5) + theme_bw() + scale_color_manual(values = derive) + theme(plot.title = element_text(hjust = 0.5)) + ggtitle("Atrial fibrillation") + labs(y = TeX('$\\hat{\\beta}$'), color = "Risk Derived") + ylim(0,max(abs(AF$EFFECT))+0.01) + theme(plot.title = element_text(hjust = 0.5),  legend.position = "bottom")
```

No HLA Hits  

Mean/Median RAF
```{r, warning=F,echo=FALSE}
mean(AF$RAF)
median(AF$RAF)
```

### Coronary Artery Disease  

Manhattan Plot - need to find the right one 

RAF vs Effect Size 
```{r, warning=FALSE, echo=FALSE}
CAD <- fread("../output/GWAS_ATLAS/evo_added/CAD_3925_evo.txt")
colnames(CAD) <- c(colnames(CAD)[1:13], "EA_Derived", "DAF", "Risk_Derived")
CAD$disease <- "Coronary artery disease"
clumped <- fread("../output/GWAS_ATLAS/clumped/CAD_3925_SNPs.txt")
CAD <- left_join(clumped, CAD, by= "SNP") 
CAD$disease <- "Coronary Artery Disease"

CAD <- subset(CAD, CAD$Risk_Derived == 'T' | CAD$Risk_Derived == 'F')
ggplot(data=CAD, aes(x=RAF,y=abs(`Log(OR)`),color=Risk_Derived)) + geom_point(alpha=0.5) + theme_bw() + scale_color_manual(values = derive) + theme(plot.title = element_text(hjust = 0.5)) + ggtitle("Coronary Artery Disease") + labs(y = TeX('$\\hat{\\beta}$'), color = "Risk Derived") + ylim(0,max(abs(CAD$`Log(OR)`))+0.01) + theme(plot.title = element_text(hjust = 0.5),  legend.position = "bottom")
```

Mean/Median RAF 
```{r, warning=F,echo=FALSE}
mean(CAD$RAF)
median(CAD$RAF)
```

Remove HLA region
```{r, warning=F,echo=FALSE}
dat <- remove_HLA(CAD)
ggplot(data=dat, aes(x=RAF,y=abs(`Log(OR)`),color=Risk_Derived)) + geom_point(alpha=0.5) + theme_bw() + scale_color_manual(values = derive) + theme(plot.title = element_text(hjust = 0.5)) + ggtitle("Coronary Artery Disease") + labs(y = TeX('$\\hat{\\beta}$'), color = "Risk Derived") + ylim(0,max(abs(dat$`Log(OR)`))+0.01) + theme(plot.title = element_text(hjust = 0.5),  legend.position = "bottom")
```

Mean/Median RAF 
```{r, warning=F,echo=FALSE}
mean(dat$RAF)
median(dat$RAF)
```

### Type 2 Diabetes  

Manhattan Plot 
```{r, out.height="100%", out.width="100%", echo=FALSE}
knitr::include_graphics("assets/T2D_4085.png", error = FALSE)
```

RAF vs Effect Size
```{r, warning=FALSE,echo=FALSE}
T2D <- fread("../output/GWAS_ATLAS/evo_added/T2D_4085_evo.txt")
colnames(T2D) <- c(colnames(T2D)[1:13], "EA_Derived", "DAF", "Risk_Derived")
T2D$disease <- "Type 2 Diabetes"
clumped <- fread("../output/GWAS_ATLAS/clumped/T2D_4085_SNPs.txt")
T2D <- left_join(clumped, T2D, by= "SNP") 

T2D <- subset(T2D, T2D$Risk_Derived == 'T' | T2D$Risk_Derived == 'F')
ggplot(data=T2D, aes(x=RAF,y=abs(BETA),color=Risk_Derived)) + geom_point(alpha=0.5) + theme_bw() + scale_color_manual(values = derive) + theme(plot.title = element_text(hjust = 0.5)) + ggtitle("Type 2 Diabetes") + labs(y = TeX('$\\hat{\\beta}$'), color = "Risk Derived") + ylim(0,max(abs(T2D$BETA))+0.01) + theme(plot.title = element_text(hjust = 0.5),  legend.position = "bottom")
```

Mean/Meadian RAF
```{r, warning=F,echo=FALSE}
mean(T2D$RAF)
median(T2D$RAF)
```

Remove HLA region
```{r, warning=F,echo=FALSE}
dat <- remove_HLA(T2D)
ggplot(data=dat, aes(x=RAF,y=abs(BETA),color=Risk_Derived)) + geom_point(alpha=0.5) + theme_bw() + scale_color_manual(values = derive) + theme(plot.title = element_text(hjust = 0.5)) + ggtitle("Type 2 Diabetes") + labs(y = TeX('$\\hat{\\beta}$'), color = "Risk Derived") + ylim(0,max(abs(dat$BETA))+0.01) + theme(plot.title = element_text(hjust = 0.5),  legend.position = "bottom")
```

Mean/Meadian RAF
```{r, warning=F,echo=FALSE}
mean(dat$RAF)
median(dat$RAF)
```


### Crohn's Disease  

Manhattan Plot  
```{r, out.height="100%", out.width="100%", echo=FALSE}
knitr::include_graphics("assets/manhattan_atlas_2029.png", error = FALSE)
```

RAF vs Effect Size
```{r, warning=F, echo=FALSE}
crohns <- fread("../output/GWAS_ATLAS/evo_added/crohns_2029_evo.txt")
colnames(crohns) <- c(colnames(crohns)[1:13], "EA_Derived", "DAF", "Risk_Derived")
crohns$disease <- "Crohns"
clumped <- fread("../output/GWAS_ATLAS/clumped/crohns_2029_SNPs.txt")
crohns <- left_join(clumped, crohns, by= "SNP") 

crohns <- subset(crohns, crohns$Risk_Derived == 'T' | crohns$Risk_Derived == 'F')
ggplot(data=crohns, aes(x=RAF,y=abs(BETA),color=Risk_Derived)) + geom_point(alpha=0.5) + theme_bw() + scale_color_manual(values = derive) + theme(plot.title = element_text(hjust = 0.5), legend.position = "bottom") + ggtitle("Crohn's Disease") + labs(y = TeX('$\\hat{\\beta}$'), color = "Risk Derived") 
```

Mean/Median RAF  
```{r, warning=F,echo=FALSE}
mean(crohns$RAF)
median(crohns$RAF)
```

Remove HLA region
```{r, warning=F,echo=FALSE}
dat <- remove_HLA(crohns)
ggplot(data=dat, aes(x=RAF,y=abs(BETA),color=Risk_Derived)) + geom_point(alpha=0.5) + theme_bw() + scale_color_manual(values = derive) + theme(plot.title = element_text(hjust = 0.5), legend.position = "bottom") + ggtitle("Crohn's Disease") + labs(y = TeX('$\\hat{\\beta}$'), color = "Risk Derived") 
```

Mean/Median RAF  
```{r, warning=F,echo=FALSE}
mean(dat$RAF)
median(dat$RAF)
```

### Breast Cancer  

Manhattan Plot  
```{r, out.height="100%", out.width="100%", echo=FALSE}
knitr::include_graphics("assets/manhattan_atlas_2029.png", error = FALSE)
```

RAF vs Effect Size  
```{r, warning=FALSE,echo=FALSE}
bc <- fread("../output/GWAS_ATLAS/evo_added/breastcancer_3659_evo.txt")
colnames(bc) <- c(colnames(bc)[1:13], "EA_Derived", "DAF", "Risk_Derived")
bc$disease <- "Breast Cancer"

bc <- subset(bc, bc$Risk_Derived == 'T' | bc$Risk_Derived == 'F')
ggplot(data=bc, aes(x=RAF,y=abs(log(OR)),color=Risk_Derived)) + geom_point(alpha=0.5) + theme_bw() + scale_color_manual(values = derive) + theme(plot.title = element_text(hjust = 0.5)) + ggtitle("Breast Cancer") + labs(y = TeX('$\\hat{\\beta}$'), color = "Risk Derived") + ylim(0,max(abs(log(bc$OR)))+0.01) + theme(plot.title = element_text(hjust = 0.5),  legend.position = "bottom")
```

Mean/Median RAF  
```{r, warning=F,echo=FALSE}
mean(bc$RAF)
median(bc$RAF)
```

Remove HLA  
```{r, warning=F,echo=FALSE}
dat <- remove_HLA(bc)
ggplot(data=dat, aes(x=RAF,y=abs(log(OR)),color=Risk_Derived)) + geom_point(alpha=0.5) + theme_bw() + scale_color_manual(values = derive) + theme(plot.title = element_text(hjust = 0.5)) + ggtitle("Breast Cancer") + labs(y = TeX('$\\hat{\\beta}$'), color = "Risk Derived") + ylim(0,max(abs(log(dat$OR)))+0.01) + theme(plot.title = element_text(hjust = 0.5),  legend.position = "bottom")
```

Mean/Median RAF  
```{r, warning=F,echo=FALSE}
mean(dat$RAF)
median(dat$RAF)
```

### Alzheimer's Disease 

Manhattan Plot  
```{r, out.height="100%", out.width="100%", echo=FALSE}
knitr::include_graphics("assets/AD_4094.png", error = FALSE)
```

RAF vs Effect Size 
```{r, warning=FALSE,echo=FALSE}
AD<- fread("../output/GWAS_ATLAS/evo_added/AD_4094_evo.txt")
colnames(AD) <- c(colnames(AD)[1:13], "EA_Derived", "DAF", "Risk_Derived")
AD$disease <- "Alzheimers"
clumped <- fread("../output/GWAS_ATLAS/clumped/AD_4094_SNPs.txt")
AD <- left_join(clumped, AD, by= "SNP") 

AD <- subset(AD, AD$Risk_Derived == 'T' | AD$Risk_Derived == 'F')
ggplot(data=AD, aes(x=RAF,y=abs(BETA),color=Risk_Derived)) + geom_point(alpha=0.5) + theme_bw() + scale_color_manual(values = derive) + theme(plot.title = element_text(hjust = 0.5)) + ggtitle("Alzheimer's Disease") + labs(y = TeX('$\\hat{\\beta}$'), color = "Risk Derived") + ylim(0,max(abs(AD$BETA))+0.01) + theme(plot.title = element_text(hjust = 0.5),  legend.position = "bottom")
```

Mean/Median RAF 
```{r, warning=F,echo=FALSE}
mean(AD$RAF)
median(AD$RAF)
```

Remove HLA 
```{r, warning=F,echo=FALSE}
dat <- remove_HLA(AD)
ggplot(data=dat, aes(x=RAF,y=abs(BETA),color=Risk_Derived)) + geom_point(alpha=0.5) + theme_bw() + scale_color_manual(values = derive) + theme(plot.title = element_text(hjust = 0.5)) + ggtitle("Alzheimer's Disease") + labs(y = TeX('$\\hat{\\beta}$'), color = "Risk Derived") + ylim(0,max(abs(dat$BETA))+0.01) + theme(plot.title = element_text(hjust = 0.5),  legend.position = "bottom")
```

Mean/Median RAF 
```{r, warning=F,echo=FALSE}
mean(dat$RAF)
median(dat$RAF)
```

### Parkinson's Disease 

Manhattan Plot  
```{r, out.height="100%", out.width="100%", echo=FALSE}
knitr::include_graphics("assets/manhattan_atlas_4167.png", error = FALSE)
```

RAF vs Effect Size
```{r, warning=FALSE,echo=FALSE}
PD<- fread("../output/GWAS_ATLAS/evo_added/parkinson_4167_evo.txt")
colnames(PD) <- c(colnames(PD)[1:13], "EA_Derived", "DAF", "Risk_Derived")
PD$disease <- "Parkinson's disease"
clumped <- fread("../output/GWAS_ATLAS/clumped/parkinson_4167_SNPs.txt")
PD <- left_join(clumped, PD, by= "SNP") 

PD <- subset(PD, PD$Risk_Derived == 'T' | PD$Risk_Derived == 'F')
ggplot(data=PD, aes(x=RAF,y=abs(BETA),color=Risk_Derived)) + geom_point(alpha=0.5) + theme_bw() + scale_color_manual(values = derive) + theme(plot.title = element_text(hjust = 0.5)) + ggtitle("Parkinson's Disease") + labs(y = TeX('$\\hat{\\beta}$'), color = "Risk Derived") + ylim(0,max(abs(PD$BETA))+0.01) + theme(plot.title = element_text(hjust = 0.5),  legend.position = "bottom")
```

Mean/Median RAF 
```{r, warning=F,echo=FALSE}
mean(PD$RAF)
median(PD$RAF)
```

Remove HLA 
```{r, warning=F,echo=FALSE}
dat <- remove_HLA(PD)
ggplot(data=dat, aes(x=RAF,y=abs(BETA),color=Risk_Derived)) + geom_point(alpha=0.5) + theme_bw() + scale_color_manual(values = derive) + theme(plot.title = element_text(hjust = 0.5)) + ggtitle("Parkinson's Disease") + labs(y = TeX('$\\hat{\\beta}$'), color = "Risk Derived") + ylim(0,max(abs(dat$BETA))+0.01) + theme(plot.title = element_text(hjust = 0.5),  legend.position = "bottom")
```

Mean/Median RAF 
```{r, warning=F,echo=FALSE}
mean(dat$RAF)
median(dat$RAF)
```

### IBD  

Manhattan Plot  
```{r, out.height="100%", out.width="100%", echo=FALSE}
knitr::include_graphics("assets/manhattan_atlas_2031.png", error = FALSE)
```

RAF vs Effect Size
```{r, warning=F, echo=FALSE}
IBD<- fread("../output/GWAS_ATLAS/evo_added/IBD_2031_evo.txt")
colnames(IBD) <- c(colnames(IBD)[1:13], "EA_Derived", "DAF", "Risk_Derived")
IBD$disease <- "Inflamatory Bowel Disease"
clumped <- fread("../output/GWAS_ATLAS/clumped/IBD_2031_SNPs.txt")
IBD <- left_join(clumped, IBD, by= "SNP") 

IBD <- subset(IBD, IBD$Risk_Derived == 'T' | IBD$Risk_Derived == 'F')
ggplot(data=IBD, aes(x=RAF,y=abs(BETA),color=Risk_Derived)) + geom_point(alpha=0.5) + theme_bw() + scale_color_manual(values = derive) + theme(plot.title = element_text(hjust = 0.5)) + ggtitle("Inflamatory Bowel Disease") + labs(y = TeX('$\\hat{\\beta}$'), color = "Risk Derived") + ylim(0,max(abs(IBD$BETA))+0.01) + theme(plot.title = element_text(hjust = 0.5),  legend.position = "bottom")
```

Mean/Median RAF 
```{r, warning=F,echo=FALSE}
mean(IBD$RAF)
median(IBD$RAF)
```

Remove HLA 
```{r, warning=F,echo=FALSE}
dat <- remove_HLA(IBD)
ggplot(data=dat, aes(x=RAF,y=abs(BETA),color=Risk_Derived)) + geom_point(alpha=0.5) + theme_bw() + scale_color_manual(values = derive) + theme(plot.title = element_text(hjust = 0.5)) + ggtitle("Inflamatory Bowel Disease") + labs(y = TeX('$\\hat{\\beta}$'), color = "Risk Derived") + ylim(0,max(abs(dat$BETA))+0.01) + theme(plot.title = element_text(hjust = 0.5),  legend.position = "bottom")
```

Mean/Median RAF 
```{r, warning=F,echo=FALSE}
mean(dat$RAF)
median(dat$RAF)
```

### Ulcerative Colitis

Manhattan Plot  
```{r, out.height="100%", out.width="100%", echo=FALSE}
knitr::include_graphics("assets/manhattan_atlas_2030.png", error = FALSE)
```

RAF vs Effect Size
```{r, warning=F,echo=FALSE}
UC<- fread("../output/GWAS_ATLAS/evo_added/UC_2030_evo.txt")
colnames(UC) <- c(colnames(UC)[1:13], "EA_Derived", "DAF", "Risk_Derived")
UC$disease <- "Ulcerative Colitis"
clumped <- fread("../output/GWAS_ATLAS/clumped/UC_2030_SNPs.txt")
UC <- left_join(clumped, UC, by= "SNP") 

UC <- subset(UC, UC$Risk_Derived == 'T' | UC$Risk_Derived == 'F')
ggplot(data=UC, aes(x=RAF,y=abs(BETA),color=Risk_Derived)) + geom_point(alpha=0.5) + theme_bw() + scale_color_manual(values = derive) + theme(plot.title = element_text(hjust = 0.5)) + ggtitle("Ulcerative Colitis") + labs(y = TeX('$\\hat{\\beta}$'), color = "Risk Derived") + ylim(0,max(abs(UC$BETA))+0.01) + theme(plot.title = element_text(hjust = 0.5),  legend.position = "bottom")

ggplot(data=UC, aes(x=RAF,y=abs(BETA))) + geom_point(alpha=0.5) + theme_bw() + theme(plot.title = element_text(hjust = 0.5)) + ggtitle("Ulcerative Colitis") + labs(y = TeX('$\\hat{\\beta}$'), color = "Risk Derived") + ylim(0,max(abs(UC$BETA))+0.01) + theme(plot.title = element_text(hjust = 0.5),  legend.position = "bottom")
```

Mean/Median RAF
```{r, warning=F,echo=FALSE}
mean(UC$RAF)
median(UC$RAF)
```

Remove HLA
```{r, warning=F,echo=FALSE}
dat <- remove_HLA(UC)
ggplot(data=dat, aes(x=RAF,y=abs(BETA),color=Risk_Derived)) + geom_point(alpha=0.5) + theme_bw() + scale_color_manual(values = derive) + theme(plot.title = element_text(hjust = 0.5)) + ggtitle("Ulcerative Colitis") + labs(y = TeX('$\\hat{\\beta}$'), color = "Risk Derived") + ylim(0,max(abs(dat$BETA))+0.01) + theme(plot.title = element_text(hjust = 0.5),  legend.position = "bottom")

ggplot(data=dat, aes(x=RAF,y=abs(BETA))) + geom_point(alpha=0.5) + theme_bw() + theme(plot.title = element_text(hjust = 0.5)) + ggtitle("Ulcerative Colitis") + labs(y = TeX('$\\hat{\\beta}$'), color = "Risk Derived") + ylim(0,max(abs(dat$BETA))+0.01) + theme(plot.title = element_text(hjust = 0.5),  legend.position = "bottom")
```

Mean/Median RAF
```{r, warning=F,echo=FALSE}
mean(dat$RAF)
median(dat$RAF)
```

### Childhood Onset Asthma

Manhattan Plot  
```{r, out.height="100%", out.width="100%", echo=FALSE}
knitr::include_graphics("assets/manhattan_atlas_4266.png", error = FALSE)
```

RAF vs Effect Size
```{r, warning=F,echo=FALSE}
CO<- fread("../output/GWAS_ATLAS/evo_added/COasthma_4266_evo.txt")
colnames(CO) <- c(colnames(CO)[1:13], "EA_Derived", "DAF", "Risk_Derived")
CO$disease <- "Childhood Onset Asthma"
clumped <- fread("../output/GWAS_ATLAS/clumped/COasthma_4266_SNPs.txt")
CO <- left_join(clumped, CO, by= "SNP") 

CO <- subset(CO, CO$Risk_Derived == 'T' | CO$Risk_Derived == 'F')
ggplot(data=CO, aes(x=RAF,y=abs(log(OR)),color=Risk_Derived)) + geom_point(alpha=0.5) + theme_bw() + scale_color_manual(values = derive) + theme(plot.title = element_text(hjust = 0.5)) + ggtitle("Childhood Onset Asthma") + labs(y = TeX('$\\hat{\\beta}$'), color = "Risk Derived") + ylim(0,max(abs(log(CO$OR)))+0.01) + theme(plot.title = element_text(hjust = 0.5),  legend.position = "bottom")
```

Mean/Median RAF 
```{r, echo=FALSE}
mean(CO$RAF)
median(CO$RAF)
```

Remove HLA
```{r, warning=F,echo=FALSE}
dat <- remove_HLA(CO)
ggplot(data=dat, aes(x=RAF,y=abs(log(OR)),color=Risk_Derived)) + geom_point(alpha=0.5) + theme_bw() + scale_color_manual(values = derive) + theme(plot.title = element_text(hjust = 0.5)) + ggtitle("Childhood Onset Asthma") + labs(y = TeX('$\\hat{\\beta}$'), color = "Risk Derived") + ylim(0,max(abs(log(dat$OR)))+0.01) + theme(plot.title = element_text(hjust = 0.5),  legend.position = "bottom")
```

Mean/Median RAF 
```{r, echo=FALSE}
mean(dat$RAF)
median(dat$RAF)
```

### Adult Onset Asthma

Manhattan Plot  
```{r, out.height="100%", out.width="100%", echo=FALSE}
knitr::include_graphics("assets/manhattan_atlas_4266.png", error = FALSE)
```

RAF vs Effect Size
```{r, warning=F,echo=FALSE}
AO<- fread("../output/GWAS_ATLAS/evo_added/AOasthma_4267_evo.txt")
colnames(AO) <- c(colnames(AO)[1:13], "EA_Derived", "DAF", "Risk_Derived")
AO$disease <- "Adult Onset Asthma"
clumped <- fread("../output/GWAS_ATLAS/clumped/AOasthma_4267_SNPs.txt")
AO <- left_join(clumped, AO, by= "SNP") 

AO <- subset(AO, AO$Risk_Derived == 'T' | AO$Risk_Derived == 'F')
ggplot(data=AO, aes(x=RAF,y=abs(log(OR)),color=Risk_Derived)) + geom_point(alpha=0.5) + theme_bw() + scale_color_manual(values = derive) + theme(plot.title = element_text(hjust = 0.5)) + ggtitle("Adult Onset Asthma") + labs(y = TeX('$\\hat{\\beta}$'), color = "Risk Derived") + ylim(0,max(abs(log(AO$OR)))+0.01) + theme(plot.title = element_text(hjust = 0.5),  legend.position = "bottom")
```

Mean/Median RAF
```{r, warning=F,echo=FALSE}
mean(AO$RAF)
median(AO$RAF)
```

Remove HLA
```{r, warning=F,echo=FALSE}
dat <- remove_HLA(AO)
ggplot(data=dat, aes(x=RAF,y=abs(log(OR)),color=Risk_Derived)) + geom_point(alpha=0.5) + theme_bw() + scale_color_manual(values = derive) + theme(plot.title = element_text(hjust = 0.5)) + ggtitle("Adult Onset Asthma") + labs(y = TeX('$\\hat{\\beta}$'), color = "Risk Derived") + ylim(0,max(abs(log(dat$OR)))+0.01) + theme(plot.title = element_text(hjust = 0.5),  legend.position = "bottom")
```

Mean/Median RAF
```{r,warning=F,echo=FALSE}
mean(dat$RAF)
median(dat$RAF)
```

### Multiple Sclerosis
```{r,warning=F,echo=FALSE}
MS <- fread("../output/GWAS_ATLAS/evo_added/MS_24076602_evo.txt")
colnames(MS) <- c(colnames(MS)[1:13], "EA_Derived", "DAF", "Risk_Derived")
MS$disease <- "Multiple Sclerosis"
clumped <- fread("../output/GWAS_ATLAS/clumped/MS_24076602_SNPs.txt")
MS <- left_join(clumped, MS, by= "SNP") 

MS <- subset(MS, MS$Risk_Derived == 'T' | MS$Risk_Derived == 'F')
ggplot(data=MS, aes(x=RAF,y=abs(BETA),color=Risk_Derived)) + geom_point(alpha=0.5) + theme_bw() + scale_color_manual(values = derive) + theme(plot.title = element_text(hjust = 0.5)) + ggtitle("Multiple Sclerosis") + labs(y = TeX('$\\hat{\\beta}$'), color = "Risk Derived") + ylim(0,max(abs(MS$BETA))+0.01) + theme(plot.title = element_text(hjust = 0.5),  legend.position = "bottom")
```

Mean/Median RAF
```{r,warning=F,echo=FALSE}
mean(MS$RAF)
median(MS$RAF)
```

Remove HLA
```{r,warning=F,echo=FALSE}
dat <- remove_HLA(MS)
ggplot(data=dat, aes(x=RAF,y=abs(BETA),color=Risk_Derived)) + geom_point(alpha=0.5) + theme_bw() + scale_color_manual(values = derive) + theme(plot.title = element_text(hjust = 0.5)) + ggtitle("Multiple Sclerosis") + labs(y = TeX('$\\hat{\\beta}$'), color = "Risk Derived") + ylim(0,max(abs(dat$BETA))+0.01) + theme(plot.title = element_text(hjust = 0.5),  legend.position = "bottom")
```

Mean/Median RAF
```{r,warning=F,echo=FALSE}
mean(dat$RAF)
median(dat$RAF)
```

### Myopia  

Manhattan Plot  
```{r, out.height="100%", out.width="100%", echo=FALSE}
knitr::include_graphics("assets/manhattan_atlas_3539.png", error = FALSE)
```

RAF vs Effect Size
```{r,warning=F,echo=FALSE}
myopia<- fread("../output/GWAS_ATLAS/evo_added/myopia_3539_evo.txt")
colnames(myopia) <- c(colnames(myopia)[1:13], "EA_Derived", "DAF", "Risk_Derived")
myopia$disease <- "Myopia"
clumped <- fread("../output/GWAS_ATLAS/clumped/myopia_3539_SNPs.txt")
myopia <- left_join(clumped, myopia, by= "SNP") 

myopia <- subset(myopia, myopia$Risk_Derived == 'T' | myopia$Risk_Derived == 'F')
ggplot(data=myopia, aes(x=RAF,y=abs(log(OR)),color=Risk_Derived)) + geom_point(alpha=0.5) + theme_bw() + scale_color_manual(values = derive) + theme(plot.title = element_text(hjust = 0.5)) + ggtitle("Myopia") + labs(y = TeX('$\\hat{\\beta}$'), color = "Risk Derived") + ylim(0,max(abs(log(myopia$OR)))+0.01) + theme(plot.title = element_text(hjust = 0.5),  legend.position = "bottom")
```

Mean/Median RAF
```{r,warning=F,echo=FALSE}
mean(myopia$RAF)
median(myopia$RAF)
```

No HLA SNPs

### Educational Attainment - UKBB only 

Manhattan Plot  
```{r, out.height="100%", out.width="100%", echo=FALSE}
knitr::include_graphics("assets/manhattan_atlas_3409.png", error = FALSE)
```

RAF vs Effect Size
```{r,warning=F,echo=FALSE}
EAukbb <- fread("../output/GWAS_ATLAS/evo_added/EAukbb_3409_evo.txt")
colnames(EAukbb) <- c(colnames(EAukbb)[1:13], "EA_Derived", "DAF", "Risk_Derived")
EAukbb$disease <- "Educational Attainment UKBB"
clumped <- fread("../output/GWAS_ATLAS/clumped/EAukbb_3409_SNPs.txt")
EAukbb <- left_join(clumped, EAukbb, by= "SNP") 

EAukbb <- subset(EAukbb, EAukbb$Risk_Derived == 'T' | EAukbb$Risk_Derived == 'F')
ggplot(data=EAukbb, aes(x=RAF,y=abs(BETA),color=Risk_Derived)) + geom_point(alpha=0.5) + theme_bw() + scale_color_manual(values = derive) + theme(plot.title = element_text(hjust = 0.5)) + ggtitle("Educational Attainment - UKBB") + labs(y = TeX('$\\hat{\\beta}$'), x = "Trait increasing allele frequency", color = "Trait increasing allele derived") + ylim(0,max(abs(EAukbb$BETA))+0.01) + theme(plot.title = element_text(hjust = 0.5),  legend.position = "bottom")
```

Mean/Median RAF
```{r,warning=F,echo=FALSE}
mean(EAukbb$RAF)
median(EAukbb$RAF)
```

Remove HLA  
```{r,warning=F,echo=FALSE}
dat <- remove_HLA(EAukbb)
ggplot(data=dat, aes(x=RAF,y=abs(BETA),color=Risk_Derived)) + geom_point(alpha=0.5) + theme_bw() + scale_color_manual(values = derive) + theme(plot.title = element_text(hjust = 0.5)) + ggtitle("Educational Attainment - UKBB") + labs(y = TeX('$\\hat{\\beta}$'), x = "Trait increasing allele frequency", color = "Trait increasing allele derived") + ylim(0,max(abs(dat$BETA))+0.01) + theme(plot.title = element_text(hjust = 0.5),  legend.position = "bottom")
```

Mean/Median RAF
```{r,warning=F,echo=FALSE}
mean(dat$RAF)
median(dat$RAF)
```

### Educational Attainment - meta-analysis
```{r,warning=F,echo=FALSE}
EAmeta<- fread("../output/GWAS_ATLAS/evo_added/EAmeta_4066_evo.txt")
colnames(EAmeta) <- c(colnames(EAmeta)[1:13], "EA_Derived", "DAF", "Risk_Derived")
EAmeta$disease <- "Educational Attainment Meta"
clumped <- fread("../output/GWAS_ATLAS/clumped/EAmeta_4066_SNPs.txt")
EAmeta <- left_join(clumped, EAmeta, by= "SNP") 

EAmeta <- subset(EAmeta, EAmeta$Risk_Derived == 'T' | EAmeta$Risk_Derived == 'F')
ggplot(data=EAmeta, aes(x=RAF,y=abs(BETA),color=Risk_Derived)) + geom_point(alpha=0.5) + theme_bw() + scale_color_manual(values = derive) + theme(plot.title = element_text(hjust = 0.5)) + ggtitle("Educational Attainment - Meta-analysis") + labs(y = TeX('$\\hat{\\beta}$'), x = "Trait increasing allele frequency", color = "Trait increasing allele derived") + ylim(0,max(abs(EAmeta$BETA))+0.01) + theme(plot.title = element_text(hjust = 0.5),  legend.position = "bottom")
```

Mean/Median RAF
```{r,warning=F,echo=FALSE}
mean(EAmeta$RAF)
median(EAmeta$RAF)
```

Remove HLA
```{r,warning=F,echo=FALSE}
dat <- remove_HLA(EAmeta)
ggplot(data=dat, aes(x=RAF,y=abs(BETA),color=Risk_Derived)) + geom_point(alpha=0.5) + theme_bw() + scale_color_manual(values = derive) + theme(plot.title = element_text(hjust = 0.5)) + ggtitle("Educational Attainment - Meta-analysis") + labs(y = TeX('$\\hat{\\beta}$'), x = "Trait increasing allele frequency", color = "Trait increasing allele derived") + ylim(0,max(abs(dat$BETA))+0.01) + theme(plot.title = element_text(hjust = 0.5),  legend.position = "bottom")
```

Mean/Median RAF
```{r,warning=F,echo=FALSE}
mean(dat$RAF)
median(dat$RAF)
```

### BMI
```{r,warning=F,echo=FALSE}
BMI <- fread("../output/GWAS_ATLAS/evo_added/BMI_4074_evo.txt")
colnames(BMI) <- c(colnames(BMI)[1:13], "EA_Derived", "DAF", "Risk_Derived")
BMI$disease <- "BMI"
clumped <- fread("../output/GWAS_ATLAS/clumped/BMI_4074.clumped")
BMI <- left_join(clumped, BMI, by= "SNP") 

BMI <- subset(BMI, BMI$Risk_Derived == 'T' | BMI$Risk_Derived == 'F')
ggplot(data=BMI, aes(x=RAF,y=abs(BETA),color=Risk_Derived)) + geom_point(alpha=0.5) + theme_bw() + scale_color_manual(values = derive) + theme(plot.title = element_text(hjust = 0.5)) + ggtitle("BMI") + labs(y = TeX('$\\hat{\\beta}$'), x = "Trait increasing allele frequency", color = "Trait increasing allele derived") + ylim(0,max(abs(BMI$BETA))+0.01) + theme(plot.title = element_text(hjust = 0.5),  legend.position = "bottom")
```

Mean/Median RAF
```{r,warning=F,echo=FALSE}
mean(BMI$RAF)
median(BMI$RAF)
```
No HLA SNPs

### PBC

Manhattan Plot  
```{r, out.height="100%", out.width="100%", echo=FALSE}
knitr::include_graphics("assets/manhattan_atlas_3822.png", error = FALSE)
```

RAF vs Effect Size
```{r,warning=F,echo=FALSE}
PBC <- fread("../output/GWAS_ATLAS/evo_added/PBC_3822_evo.txt")
colnames(PBC) <- c(colnames(PBC)[1:13], "EA_Derived", "DAF", "Risk_Derived")
PBC$disease <- "Primary Biliary Cholangitis"
clumped <- fread("../output/GWAS_ATLAS/clumped/PBC_3822_SNPs.txt")
PBC <- left_join(clumped, PBC, by= "SNP") 

PBC <- subset(PBC, PBC$Risk_Derived == 'T' | PBC$Risk_Derived == 'F')
ggplot(data=PBC, aes(x=RAF,y=abs(BETA),color=Risk_Derived)) + geom_point(alpha=0.5) + theme_bw() + scale_color_manual(values = derive) + theme(plot.title = element_text(hjust = 0.5)) + ggtitle("Primary Biliary Cholangitis") + labs(y = TeX('$\\hat{\\beta}$'), color = "Risk Derived") + ylim(0,max(abs(PBC$BETA))+0.01) + theme(plot.title = element_text(hjust = 0.5),  legend.position = "bottom")
```

Mean/Median RAF
```{r,warning=F,echo=FALSE}
mean(PBC$RAF)
median(PBC$RAF)
```

Remove HLA
```{r,warning=F,echo=FALSE}
dat <- remove_HLA(PBC)
ggplot(data=dat, aes(x=RAF,y=abs(BETA),color=Risk_Derived)) + geom_point(alpha=0.5) + theme_bw() + scale_color_manual(values = derive) + theme(plot.title = element_text(hjust = 0.5)) + ggtitle("Primary Biliary Cholangitis") + labs(y = TeX('$\\hat{\\beta}$'), color = "Risk Derived") + ylim(0,max(abs(dat$BETA))+0.01) + theme(plot.title = element_text(hjust = 0.5),  legend.position = "bottom")
```

Mean/Median RAF
```{r,warning=F,echo=FALSE}
mean(dat$RAF)
median(dat$RAF)
```

### Vitiligo 

Manhattan Plot  
```{r, out.height="100%", out.width="100%", echo=FALSE}
knitr::include_graphics("assets/manhattan_atlas_3822.png", error = FALSE)
```

RAF vs Effect Size
```{r, warning=F, echo=FALSE}
vit <- fread("../output/GWAS_ATLAS/evo_added/vitiligo_3832_evo.txt")
colnames(vit) <- c(colnames(vit)[1:13], "EA_Derived", "DAF", "Risk_Derived")
vit$disease <- "Vitiligo"
clumped <- fread("../output/GWAS_ATLAS/clumped/vitiligo_3832_SNPs.txt")
vit <- left_join(clumped, vit, by= "SNP") 

vit <- subset(vit, vit$Risk_Derived == 'T' | vit$Risk_Derived == 'F')
ggplot(data=vit, aes(x=RAF,y=abs(log(OR)),color=Risk_Derived)) + geom_point(alpha=0.5) + theme_bw() + scale_color_manual(values = derive) + theme(plot.title = element_text(hjust = 0.5)) + ggtitle("Vitiligo") + labs(y = TeX('$\\hat{\\beta}$'), color = "Risk Derived") + ylim(0,max(abs(log(vit$OR)))+0.01) + theme(plot.title = element_text(hjust = 0.5),  legend.position = "bottom")
```

Mean/Median RAF
```{r,warning=F,echo=FALSE}
mean(vit$RAF)
median(vit$RAF)
```

Remove HLA
```{r,warning=F,echo=FALSE}
dat <- remove_HLA(vit)
ggplot(data=dat, aes(x=RAF,y=abs(log(OR)),color=Risk_Derived)) + geom_point(alpha=0.5) + theme_bw() + scale_color_manual(values = derive) + theme(plot.title = element_text(hjust = 0.5)) + ggtitle("Vitiligo") + labs(y = TeX('$\\hat{\\beta}$'), color = "Risk Derived") + ylim(0,max(abs(log(dat$OR)))+0.01) + theme(plot.title = element_text(hjust = 0.5),  legend.position = "bottom")
```

Mean/Median RAF
```{r,warning=F,echo=FALSE}
mean(dat$RAF)
median(dat$RAF)
```

### Bipolar Disorder 

RAF vs Effect Size 
```{r, warning=F, echo=FALSE}
BIP <- fread("../output/GWAS_ATLAS/evo_added/BIP_4039_evo.txt")
colnames(BIP) <- c(colnames(BIP)[1:13], "EA_Derived", "DAF", "Risk_Derived")
BIP$disease <- "Bipolar Disorder"
clumped <- fread("../output/GWAS_ATLAS/clumped/BIP_4039_SNPs.txt")
BIP <- left_join(clumped, BIP, by= "SNP") 

BIP <- subset(BIP, BIP$Risk_Derived == 'T' | BIP$Risk_Derived == 'F')
ggplot(data=BIP, aes(x=RAF,y=abs(log(OR)),color=Risk_Derived)) + geom_point(alpha=0.5) + theme_bw() + scale_color_manual(values = derive) + theme(plot.title = element_text(hjust = 0.5)) + ggtitle("Bipolar Disorder") + labs(y = TeX('$\\hat{\\beta}$'), color = "Risk Derived") + ylim(0,max(abs(log(BIP$OR)))+0.01) + theme(plot.title = element_text(hjust = 0.5),  legend.position = "bottom")
```

Mean/Median RAF
```{r,warning=F,echo=FALSE}
mean(BIP$RAF)
median(BIP$RAF)
```

No HLA SNPs 


## Median RAF 
```{r,warning=F,echo=FALSE}
combined <- bind_rows(AD,AF,AO,bc,BIP,BMI,CAD,CKD,CO,crohns,EAmeta, EAukbb,IBD,MS,myopia,PBC,PD,schizophrenia,scz_pgc,T2D,UC,vit)

fig4 <- combined %>% select(RAF, disease) %>% na.omit()
meds <- fig4 %>% group_by(disease) %>% summarise(med = median(RAF, na.rm = T))
ord <- meds[order(meds$med),]
fig4$disease <- as.factor(fig4$disease)
fig4$disease <- factor(fig4$disease, levels = ord$disease)

ggplot(data=fig4, aes(x=disease , y=RAF)) + geom_boxplot(col = "navy", fill = "lightblue") + coord_flip() + ylab("Risk Allele Frequency") + xlab("Trait") + theme_bw()

#ggsave("~/Desktop/boxplots.png",pl, height = 5, width = 9)
```





```{r, echo=FALSE}
# schizophrenia <- fread("../output/GWAS_ATLAS/evo_added/schizophrenia_3982_evo.txt")
# colnames(schizophrenia) <- c(colnames(schizophrenia)[1:13], "EA_Derived", "DAF", "Risk_Derived")
# schizophrenia$disease <- "Schizophenria"
# 
# EAmeta<- fread("../output/GWAS_ATLAS/evo_added/EAmeta_4066_evo.txt")
# colnames(EAmeta) <- c(colnames(EAmeta)[1:13], "EA_Derived", "DAF", "Risk_Derived")
# EAmeta$disease <- "Educational Attainment Meta"
# 
# olap <- inner_join(schizophrenia, EAmeta, by = 'SNP')
# ggplot(data=olap, aes(x=RAF.x,y=abs(log(OR)),color=Risk_Derived.x)) + geom_point(alpha=0.5) + theme_bw() + scale_color_manual(values = derive) + theme(plot.title = element_text(hjust = 0.5)) + ggtitle("Risk Beta, colored by D/A") 

```




```{r,echo=FALSE}
# good <- fread("~/Downloads/Within-family_GWAS_of_height_based_on_sib_regression_using_data_from_Robinson_et_al_2015_LYMRR.txt")
# 
# top_good <- good %>% filter(PVAL <= 1e-3)
# risk <- subset(top_good, top_good$BETA > 0)
# risk$RAF <- risk$FREQ_A1
# protective <- subset(top_good, top_good$BETA < 0)
# protective$RAF <- 1 - protective$FREQ_A1
# dat <- rbind(risk, protective)
# 
# plot(dat$RAF, abs(dat$BETA))
# plot(top_good$FREQ_A1, top_good$BETA)
```

```{r,echo=F}
# strat <- fread("~/Downloads/withinfam_summary_ht_bmi_release_March2016/qfam_sibpair_height_March2016.ma")
# 
# top <- strat %>% filter(p >= 0.999)
# risk <- subset(top, top$b > 0)
# risk$RAF <- risk$freq
# protective <- subset(top, top$b < 0)
# protective$RAF <- 1 - protective$freq
# dat <- rbind(risk, protective)
# 
# plot(dat$RAF, abs(dat$b))
# plot(top$freq, top$b)
```


