---
title: "misc_gwas"
author: "Jennifer Blanc"
date: "6/8/2020"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
rm(list =ls())
library(data.table)
library(tidyverse)
library(knitr)
library(ggpubr)
```

## Code  

Read in summary statistics  
```{r}
read_ss <- function(name) {
  df <- fread(paste0("../data/MISC_GWAS/",name, ".txt"))
  return(df)
}
```

Pick only GWAS Significant SNPs 
```{r}
sig_snps <- function(df, threshold) {
  out <- df %>% filter(P < threshold)
  return(out)
}
```

Pick lowest p-value SNP per block 
```{r}
lowest_pval <- function(df) {
  out <- df %>% group_by(LD_BLOCK) %>% arrange(P) %>% slice(1)
  return(out)
}
```

Pick SNPs below a given threshold  
```{r}
nonsig_snps <- function(df, threshold) {
  out <- df %>% filter(abs(ES) < threshold)
}
```

Proportion of derived alleles that increase the trait  
```{r}
increasing_derived_alleles <- function(df) {
  return(sum(df$ES_DERIVED > 0)/nrow(df))
}
```

Proportion of derived alleles that increase the trait for multiple thresholds  
```{r}
increasing_derived_alleles_thresolds <- function(data) {
  df <- matrix(NA, nrow = 5, ncol = 2)
  colnames(df) <- c("Prop_Inc", "Num_SNPs")
  threshold <- c(1e-2, 1e-3, 1e-4, 1e-5, 1e-6)
  for (i in 1:5) {
    non_sig <- nonsig_snps(data, threshold[i])
    df[i,1] <- increasing_derived_alleles(non_sig)
    df[i,2] <- nrow(non_sig)
    data <- non_sig
  }
  return(df)
}
```


Make ES vs IAF plot
```{r}
derive <- c("springgreen4", "purple1")
make_plot <- function(df, name) {
  pl <- ggplot(data=df, aes(x=IAF,y=abs(ES), color = INCREASING_DERIVED)) + geom_point(alpha=0.5, size=2) + theme_classic() + ggtitle(name) + labs(y = "Effect Size", x = "Trait Increaseing Allele Frequency",color = "Risk Derived")+ ylim(0,max(abs(df$ES)+0.01)) +  xlim(0,1) + scale_color_manual(values = derive) + theme(plot.title = element_text(hjust = 0.5, size = 20), axis.text.x = element_text(size = 10), axis.text.y = element_text(size = 10), axis.title.x = element_text(size = 18), axis.title.y = element_text(size = 18), legend.title = element_text(size = 14), legend.text = element_text(size = 14), legend.position = "bottom") 
  return(pl)
}
```


## Workflow outline  

1) Download raw summary statistics  
2) Parse all summary stats into same format: "SNP" (CHR:BP), "EA" (effect allele),"NEA"(non-effect allele), "CHR", "BP", "BETA" (Should be called OR-need to fix), "SE", "P", "EAF" (effect allele frequency) ,"MAF", "IAF" (increasing allele frequency), "INCREASING_ALLELE" (BETA > 0), "INCREASING" (Allele w/ BETA > 0), "ES" (log(OR))
3) Add column that uses a table that has derived/ancestral assignments for 1KG SNPs to add the columns: "EA_DERIVED" (T/F effect allele derived), "DAF" (derived allele freq), "INCREASING_DERIVED" (T/F is the increasing allele derived), "ES_DERIVED" (Effect size of the derived allele). SNPs that don't have derived information are dropped from the table.    
4) Add column that assigns each SNP to one of 1703 LD Blocks defined by https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4731402/  


## Schizophrenia GWAS Results 

**EUR Psychiatric Genetic Consortium**  

The raw GWAS was downloaded from (https://www.med.unc.edu/pgc/download-results/scz/), EUR only descent. They are discussed in this paper: https://www.nature.com/articles/s41588-019-0512-x   

First, we can plot the trait increasing allele frequency vs effect size for GWAS significant SNPs (P < 5e-8), picking the lowest p-value per LD Block and coloring by derived status.    
```{r}
data <- fread("~/scratch/SCZ-EUR_PGC.txt")
sig <- sig_snps(data, 5e-8)
ld_block <- lowest_pval(sig)
make_plot(ld_block, "SCZ-EUR_PGC")
```

Now we can make the same plot but dropping all SNPs on chromosome 6.  
```{r}
drop_six <- ld_block %>% filter(CHR != 6)
make_plot(drop_six, "Exclude CHR 6")
```

We can also look at specific LD Blocks on chromosome 6  
```{r}
p650 <- data %>% filter(CHR == 6 & LD_BLOCK == 650) %>% mutate(sig = P < 5e-8) %>%ggplot(aes(x = BP, y=ES_DERIVED, color = sig)) + geom_point()
p651 <- data %>% filter(CHR == 6 & LD_BLOCK == 651) %>% mutate(sig = P < 5e-8) %>%ggplot(aes(x = BP, y=ES_DERIVED, color = sig)) + geom_point()
p652 <- data %>% filter(CHR == 6 & LD_BLOCK == 652) %>% mutate(sig = P < 5e-8) %>%ggplot(aes(x = BP, y=ES_DERIVED, color = sig)) + geom_point()
p653 <- data %>% filter(CHR == 6 & LD_BLOCK == 653) %>% mutate(sig = P < 5e-8) %>%ggplot(aes(x = BP, y=ES_DERIVED, color = sig)) + geom_point()
ggarrange(p650,p651,p652,p653)
```


```{r}
#df <- data %>% mutate(daf_bin = cut(DAF,c(-Inf,0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9, Inf), labels = seq(0.1,1, 0.1))) %>% mutate(es_bin = cut(ES_DERIVED,c(-Inf,-0.7, -0.6, -0.5, -0.4, -0.3, -0.2, -0.1, 0, 0.1,0.2,0.3,0.4,0.5,0.6,0.7, Inf), labels = c(-0.7, -0.6, -0.5, -0.4, -0.3, -0.2, -0.1, 0, 0.1,0.2,0.3,0.4,0.5,0.6,0.7,0.8)))
#df_bin <- df %>% group_by(daf_bin, es_bin) %>% summarise(num_snps = n())
#ggplot(data = df_bin, aes(x=daf_bin, y=es_bin, fill = num_snps)) + geom_tile()
```



**EAS Psychiatric Genetic Consortium**  

The raw GWAS was downloaded from (https://www.med.unc.edu/pgc/download-results/scz/), EUR only descent. They are discussed in this paper: https://www.nature.com/articles/s41588-019-0512-x   

First, we can plot the trait increasing allele frequency vs effect size for GWAS significant SNPs (P < 5e-8), picking the lowest p-value per LD Block and coloring by derived status. (n=21)   
```{r}
data <- fread("~/scratch/SCZ-EAS_PGC.txt")
sig <- sig_snps(data, 5e-8)
ld_block <- lowest_pval(sig)
make_plot(ld_block, "SCZ-EAS_PGC")
```

Here is the same plot with all the significant SNPs (not picking the lowest per LD Block).  
```{r}
make_plot(sig, "SCZ-EAS_PGC, all SNPs")
```


Let's try dropping the significance threshold (P < 5e-4), still picking the lowest p-value per LD Block.  
```{r}
sig <- sig_snps(data, 5e-6)
make_plot(sig, "SCZ-EAS_PGC")
```

**EUR Walters Group Data**  

The raw GWAS was downloaded from (https://walters.psycm.cf.ac.uk/, EUR only descent. They are discussed in this paper: https://www.nature.com/articles/s41588-018-0059-2   

First, we can plot the trait increasing allele frequency vs effect size for GWAS significant SNPs (P < 5e-8), picking the lowest p-value per LD Block and coloring by derived status.    
```{r}
data <- fread("~/scratch/SCZ_3982.txt")
sig <- sig_snps(data, 5e-8)
ld_block <- lowest_pval(sig)
make_plot(ld_block, "SCZ-Walters")
```

Now we can make the same plot but dropping all SNPs on chromosome 6.  
```{r}
drop_six <- ld_block %>% filter(CHR != 6)
make_plot(drop_six, "Exclude CHR 6")
```
